---
title: "R Notebook"
output: html_notebook
---

This is an [R Markdown](http://rmarkdown.rstudio.com) Notebook. When you execute code within the notebook, the results appear beneath the code. 

Try executing this chunk by clicking the *Run* button within the chunk or by placing your cursor inside it and pressing *Ctrl+Shift+Enter*. 

```{r}
library(ggplot2)
library(DESeq2)
library(MaizePal)
library(rstatix)
library(pheatmap)
library(viridisLite)
library(viridis)
library(ggforce)
library(ggExtra)
library(gridExtra)
library(matrixStats)
library("ggalluvial")
library(dplyr)
library(reshape2)
```

```{r}
counts <- read.table("//las-dfs-01.las.iastate.edu/lss/research/sna-lab/kmh/projects/timepoint/combined_counts_W22xB73_recips_timeseries.txt",header=T,stringsAsFactors = F)

#Change row name to the gene ID
rownames(counts) <- counts$Geneid
counts$Geneid <- NULL

#Normalize counts into reads per million
counts.rpm <- counts
for(i in 1:23){
  counts.rpm[,i] <- (counts[,i]/sum(counts[,i])) * 1e6
}
##Create heatmap to see if groups cluster together
counts.rpm2 <- subset(counts.rpm, grepl("Zm|TE", rownames(counts.rpm)))
counts.rpm.3 <- counts.rpm2[,1:23]

data_subset <- as.matrix(counts.rpm.3[rowSums(counts.rpm.3)>1000,])

pheatmap((as.matrix(data_subset)/rowMaxs(as.matrix(data_subset))), border_color = NA, fontsize = 8, show_rownames = F)

##Correlation heatmap
library(NMF)
corr <- cor(counts.rpm.3)
pheatmap::pheatmap(corr,main = "Correlation beween samples",labCol=substr(colnames(counts.rpm.3),1,4),labRow=substr(colnames(counts.rpm.3),1,4))


#Subset only the counts getting rid of nonaligned or multimapping
counts2 <- subset(counts, grepl("Zm|TE", rownames(counts)))
counts.de <- counts2[,1:23]
rownames(counts.de) <- row.names(counts2)

#Get rid of rows with <10 reads in non-normalized set
counts.de.keep <- subset(counts.de,rowSums(counts.de) >= 10) 

##summary stats
counts2.numerator <- counts2
counts2.denominator <- counts2
# remove non-feature and ambiguous reads


summary.stats.libraries <- data.frame(cbind(colSums(counts2.denominator[,2:23]),(colSums(counts2.numerator[,2:23])/colSums(counts2.denominator[,2:23]))*100))
names(summary.stats.libraries) <- c("Reads","Percent_Unique")
mean(summary.stats.libraries$Percent_Unique)

#Make sure the mutant and wildtype have at least 1 read in each of the 3 reps in normalized set
#counts.de.keep <- subset(counts.de,rowMeans(counts.rpm[,1:3]) >= 0.5 | rowMeans(counts.rpm[,4:6]) >= 0.5) 

#Set sample information for DE calculation
sample.info <- as.data.frame(cbind(paste(names(counts.de.keep)),c("BW11", "BW11", "BW11", "WB11", "WB11", "WB11", "BW14", "BW14", "BW14", "WB14", "WB14", "WB14", "BW17", "BW17", "BW17", "WB17", "WB17", "WB17", "BW21", "BW21", "BW21", "WB21", "WB21")))
rownames(sample.info) <- sample.info[,1]
sample.info$timepoint <- substr(sample.info$V2,3,4)
sample.info$genotype <- substr(sample.info$V1,1,2)

write.table(sample.info, file = "//las-dfs-01.las.iastate.edu/lss/research/sna-lab/kmh/projects/timepoint/sample.info.txt", col.names = T, row.names = F, quote = F)

#Run DESeq
dds <- DESeqDataSetFromMatrix(countData = counts.de.keep,colData = sample.info, design = ~ timepoint + genotype + timepoint:genotype)
dds2 <- DESeq(dds, test = "LRT", reduced = ~ timepoint + genotype)
keep <- rowSums(counts(dds2)) >1
dds2 <- dds2[keep,]

plotDispEsts(dds2)
#Get names of different tests performed
resultsNames(dds2)
#make results tables
res_17v11 <- results(dds2, name = "timepoint_17_vs_11", alpha = .001 )
res_14v11 <- results(dds2, name = "timepoint_14_vs_11", alpha = .001 )
res_21v11 <- results(dds2, name = "timepoint_21_vs_11", alpha = .001 )
res_WBvBW <- results(dds2, name = "genotype_WB_vs_BW", alpha = .001 )
res_int <- results(dds2, name = "Intercept", alpha = .001)

res_14WB <- results(dds2, name = "timepoint14.genotypeWB")
#Coerce results to dataframes
out_17v11<- data.frame(res_17v11)
out_14v11<- data.frame(res_14v11)
out_21v11<- data.frame(res_21v11)
out_WBvBW<- data.frame(res_WBvBW)
out_int <- data.frame(res_int)
out_14WB <- data.frame(res_14WB)

rld <- rlog(dds2, blind=FALSE)
colData(rld)
PCA_time <- plotPCA(rld,intgroup="V2", ntop = 10000) + theme_classic() 
PCA_time

PCA_geno <- plotPCA(rld,intgroup="genotype", ntop = 10000)
PCA_geno + theme_classic()

#write.table(counts.rpm.3, file = "//las-dfs-01.las.iastate.edu/lss/research/sna-lab/kmh/projects/timepoint/timepoint.rpm.txt")

#Make Differential expression calls based on adjusted p-value and fold change
out_14v11$Diff_Exp <- ifelse(out_14v11$padj < .05 & out_14v11$log2FoldChange > 1, "up", ifelse(out_14v11$padj < .05 & out_14v11$log2FoldChange < -1, "down", "notDE"))
out_17v11$Diff_Exp <- ifelse(out_17v11$padj < .05 & out_17v11$log2FoldChange > 1, "up", ifelse(out_17v11$padj < .05 & out_17v11$log2FoldChange < -1, "down", "notDE"))
out_21v11$Diff_Exp <- ifelse(out_21v11$padj < .05 & out_21v11$log2FoldChange > 1, "up", ifelse(out_21v11$padj < .05 & out_21v11$log2FoldChange < -1, "down", "notDE"))
out_14v11$ID <- row.names(out_14v11)
out_17v11$ID <- row.names(out_17v11)
out_21v11$ID <- row.names(out_21v11)

#subset genes upregulated and down regulated at each timepoint



DE_14v11 <- out_14v11[,c("ID", "Diff_Exp")]
DE_14and17v11 <- left_join(DE_14v11, out_17v11[,c("ID", "Diff_Exp")], by = "ID")
colnames(DE_14and17v11) <- c("ID", "DE_14DAP", "DE 17DAP")
DE_time <-  left_join(DE_14and17v11, out_21v11[,c("ID", "Diff_Exp")], by = "ID")
colnames(DE_time) <- c("ID", "DE_14DAP", "DE 17DAP","DE 21DAP")

DE_time[is.na(DE_time)] <- "notDE"

#Summarize all columns to see major patterns and plot
sum_time <- DE_time %>% group_by(DE_14DAP,`DE 17DAP`,`DE 21DAP`) %>% dplyr::summarize(count = n())
#Plot total genes up and down-regulated at each time point
```

```{r} 
#Plot barchart of total counts DE at each timepoint
summary.14v11 <- data.frame(out_14v11 %>% group_by(Diff_Exp) %>% summarize(count = n()))
summary.17v11 <- data.frame(out_17v11 %>% group_by(Diff_Exp) %>% summarize(count = n()))
summary.21v11 <- data.frame(out_21v11 %>% group_by(Diff_Exp) %>% summarize(count = n()))

summary.combined <- bind_rows(summary.14v11,summary.17v11,summary.21v11, .id = "Contrast")
summary.combined$Contrast <- ifelse(summary.combined$Contrast == 1, "14DAP v 11DAP", ifelse(summary.combined$Contrast == 2, "17DAP v 11DAP", "21DAP v 11DAP"))

summary.combined2 <- subset(summary.combined,summary.combined$Diff_Exp %in% c("up","down"))
summary.combined2[summary.combined2$Diff_Exp == "down","count"] <- -summary.combined2[summary.combined2$Diff_Exp == "down","count"]

ggplot() + geom_bar(aes(x=Contrast,y=count,fill=Diff_Exp),data=summary.combined2,stat="identity",position = "identity") + theme_classic()  +  theme(axis.text.x = element_text(angle=90, hjust=1)) + scale_fill_manual(values=c("orange","maroon")) 

#How many of these are consistently up or down regulated?
library(ggVennDiagram)
#Subset Imprinting dfs so they are MEG or PEG
up_14 <- subset(out_14v11, out_14v11$Diff_Exp == "up")
up_17 <- subset(out_17v11, out_17v11$Diff_Exp == "up")
up_21 <- subset(out_21v11, out_21v11$Diff_Exp == "up")

down_14 <- subset(out_14v11, out_14v11$Diff_Exp == "down")
down_17 <- subset(out_17v11, out_17v11$Diff_Exp == "down")
down_21 <- subset(out_21v11, out_21v11$Diff_Exp == "down")


#Same thing, but look at ID type too
up_intersection <- list(A = up_14$ID, B = up_17$ID, C = up_21$ID)
down_intersection <- list(A = down_14$ID, B = down_17$ID, C = down_21$ID)
ggVennDiagram(up_intersection) + scale_fill_gradient(low = "gold2", high = "firebrick") + ggtitle("upregulated genes")
ggVennDiagram(down_intersection) + scale_fill_gradient(low = "aquamarine3", high = "navy") + ggtitle("downregulated genes")
```



```{r}
##Read in Cluster calls
read.table(file = "//las-dfs-01.las.iastate.edu/lss/research/sna-lab/kmh/projects/timepoint/cluster_assignment.txt", col.names = T, sep = "\t")
```


```{r}

#Get rid of low counts

counts.rpm.keep <- subset(counts.rpm.3,rowSums(counts.rpm.3) >= 10) 


counts.rpm.keep$genome <-ifelse(grepl("Zm00001e|B73",rownames(counts.rpm.keep)),"B73",ifelse(grepl("Zm00004b|W22",rownames(counts.rpm.keep)), "W22","NA")) # Label which genome the feature annotation is from
table(counts.rpm.keep$genome)


#Subset one contrast
imp.11DAP.rpm <- counts.rpm.keep[,c(1:6,24)]
#Give the contrast a name, in this case it would be the timepoint
imp.11DAP.rpm$contrast <- "11DAP"
#Create a new column to simplify using which genome the gene originated from 
imp.11DAP.rpm$type <- ifelse(imp.11DAP.rpm$genome == "B73","A","B")
#transfer rownames
imp.11DAP.rpm$feature <- rownames(imp.11DAP.rpm)

#Rename columns, reduce the 'cross' columns for comparisons with fewer columns
names(imp.11DAP.rpm) <- c("cross1","cross2","cross3", "cross4","cross5","cross6","genome","contrast","type","feature")

#Create the A_mean and B_mean, in this example B73 is the first 3 columns of the subset dataframe, so I ask for the mean of those three columns as the A_mean, and the other 3 as the B_mean
imp.11DAP.rpm$A_mean <- rowMeans(imp.11DAP.rpm[,1:3])
imp.11DAP.rpm$B_mean <- rowMeans(imp.11DAP.rpm[,4:6])

#here is the calculation of maternal preference
imp.11DAP.rpm$maternal_preference <- ifelse(imp.11DAP.rpm$type == "A",imp.11DAP.rpm$A_mean/(imp.11DAP.rpm$B_mean + imp.11DAP.rpm$A_mean) ,imp.11DAP.rpm$B_mean/(imp.11DAP.rpm$B_mean + imp.11DAP.rpm$A_mean))

#Subset one contrast
imp.14DAP.rpm <- counts.rpm.keep[,c(7:12,24)]
#Give the contrast a name, in this case it would be the timepoint
imp.14DAP.rpm$contrast <- "14DAP"
#Create a new column to simplify using which genome the gene originated from 
imp.14DAP.rpm$type <- ifelse(imp.14DAP.rpm$genome == "B73","A","B")
#transfer rownames
imp.14DAP.rpm$feature <- rownames(imp.14DAP.rpm)

#Rename columns, reduce the 'cross' columns for comparisons with fewer columns
names(imp.14DAP.rpm) <- c("cross1","cross2","cross3", "cross4","cross5","cross6","genome","contrast","type","feature")

#Create the A_mean and B_mean, in this example B73 is the first 3 columns of the subset dataframe, so I ask for the mean of those three columns as the A_mean, and the other 3 as the B_mean
imp.14DAP.rpm$A_mean <- rowMeans(imp.14DAP.rpm[,1:3])
imp.14DAP.rpm$B_mean <- rowMeans(imp.14DAP.rpm[,4:6])

#here is the calculation of maternal preference
imp.14DAP.rpm$maternal_preference <- ifelse(imp.14DAP.rpm$type == "A",imp.14DAP.rpm$A_mean/(imp.14DAP.rpm$B_mean + imp.14DAP.rpm$A_mean) ,imp.14DAP.rpm$B_mean/(imp.14DAP.rpm$B_mean + imp.14DAP.rpm$A_mean))

#Subset one contrast
imp.17DAP.rpm <- counts.rpm.keep[,c(13:18,24)]
#Give the contrast a name, in this case it would be the timepoint
imp.17DAP.rpm$contrast <- "17DAP"
#Create a new column to simplify using which genome the gene originated from 
imp.17DAP.rpm$type <- ifelse(imp.17DAP.rpm$genome == "B73","A","B")
#transfer rownames
imp.17DAP.rpm$feature <- rownames(imp.17DAP.rpm)

#Rename columns, reduce the 'cross' columns for comparisons with fewer columns
names(imp.17DAP.rpm) <- c("cross1","cross2","cross3", "cross4","cross5","cross6","genome","contrast","type","feature")

#Create the A_mean and B_mean, in this example B73 is the first 3 columns of the subset dataframe, so I ask for the mean of those three columns as the A_mean, and the other 3 as the B_mean
imp.17DAP.rpm$A_mean <- rowMeans(imp.17DAP.rpm[,1:3])
imp.17DAP.rpm$B_mean <- rowMeans(imp.17DAP.rpm[,4:6])

#here is the calculation of maternal preference
imp.17DAP.rpm$maternal_preference <- ifelse(imp.17DAP.rpm$type == "A",imp.17DAP.rpm$A_mean/(imp.17DAP.rpm$B_mean + imp.17DAP.rpm$A_mean) ,imp.17DAP.rpm$B_mean/(imp.17DAP.rpm$B_mean + imp.17DAP.rpm$A_mean))

#Subset one contrast
imp.21DAP.rpm <- counts.rpm.keep[,c(19:23,24)]
#Give the contrast a name, in this case it would be the timepoint
imp.21DAP.rpm$contrast <- "21DAP"
#Create a new column to simplify using which genome the gene originated from 
imp.21DAP.rpm$type <- ifelse(imp.21DAP.rpm$genome == "B73","A","B")
#transfer rownames
imp.21DAP.rpm$feature <- rownames(imp.21DAP.rpm)

#Rename columns, reduce the 'cross' columns for comparisons with fewer columns
names(imp.21DAP.rpm) <- c("cross1","cross2","cross3", "cross4","cross5","genome","contrast","type","feature")

#Create the A_mean and B_mean, in this example B73 is the first 3 columns of the subset dataframe, so I ask for the mean of those three columns as the A_mean, and the other 3 as the B_mean
imp.21DAP.rpm$A_mean <- rowMeans(imp.21DAP.rpm[,1:3])
imp.21DAP.rpm$B_mean <- rowMeans(imp.21DAP.rpm[,4:5])

#here is the calculation of maternal preference
imp.21DAP.rpm$maternal_preference <- ifelse(imp.21DAP.rpm$type == "A",imp.21DAP.rpm$A_mean/(imp.21DAP.rpm$B_mean + imp.21DAP.rpm$A_mean) ,imp.21DAP.rpm$B_mean/(imp.21DAP.rpm$B_mean + imp.21DAP.rpm$A_mean))

imp.21.mat <- imp.21DAP.rpm[,c("feature","maternal_preference","contrast")]
imp.17.mat <- imp.17DAP.rpm[,c("feature","maternal_preference","contrast")]
imp.14.mat <- imp.14DAP.rpm[,c("feature","maternal_preference","contrast")]
imp.11.mat <- imp.11DAP.rpm[,c("feature","maternal_preference","contrast")]


imp.mat1 <- right_join(imp.11.mat,imp.17.mat, by = "feature")
imp.mat2 <- right_join(imp.mat1,imp.14.mat, by = "feature")
imp.mat.all <- right_join(imp.mat2, imp.21.mat, by ="feature")

#rename columns
names(imp.mat.all) <- c("feature","mat_pref11","11DAP","mat_pref17","17DAP","mat_pref14","14DAP","mat_pref21","21DAP")
#prep for melting 
prep <- imp.mat.all[,c(1,2,4,6,8)]


#Make df for including mean rpms for each timepoint
imp.21.rpm <- imp.21DAP.rpm[,c("feature","A_mean","B_mean","genome","contrast")]
imp.17.rpm <- imp.17DAP.rpm[,c("feature","A_mean","B_mean","contrast")]
imp.14.rpm <- imp.14DAP.rpm[,c("feature","A_mean","B_mean","contrast")]
imp.11.rpm <- imp.11DAP.rpm[,c("feature","A_mean","B_mean","contrast")]



imp.rpm1 <- right_join(imp.11.rpm,imp.14.rpm, by = "feature")
imp.rpm2 <- right_join(imp.rpm1,imp.17.rpm, by = "feature")
imp.rpm.all <- right_join(imp.rpm2, imp.21.rpm, by ="feature")

#rename columns
names(imp.rpm.all) <- c("feature","A_mean11","B_mean11","11DAP","A_mean14","B_mean14","14DAP","A_mean17","B_mean17","17DAP","A_mean21","B_mean21","genome","21DAP")

#prep for melt
prep2 <- imp.rpm.all[,c(1,2,3,5,6,8,9,11:13)]
#melt
imp.rpm.melted <- melt(prep2, id.vars = c("feature","genome"),varnames = c("A_mean","B_mean"))
#Create new columns for coloring
imp.rpm.melted$Timepoint <- ifelse(grepl("11", imp.rpm.melted$variable), "11 DAP",ifelse(grepl("14", imp.rpm.melted$variable), "14 DAP",ifelse(grepl("17", imp.rpm.melted$variable), "17 DAP",ifelse(grepl("21", imp.rpm.melted$variable), "21 DAP", "na"))))
imp.rpm.melted$group <- ifelse(grepl("A_mean", imp.rpm.melted$variable), "A",ifelse(grepl("B_mean", imp.rpm.melted$variable), "B","na"))
imp.rpm.melted$group2 <- ifelse(imp.rpm.melted$group=="A" & imp.rpm.melted$genome == "B73", "maternal", ifelse(imp.rpm.melted$group=="B"& imp.rpm.melted$genome=="W22", "maternal", "paternal"))




imp.all.melted <- melt(prep, id.vars = "feature",value.name = "maternal_preference")

imp.all.melted$Timepoint <- ifelse(grepl("11", imp.all.melted$variable), "11 DAP",ifelse(grepl("14", imp.all.melted$variable), "14 DAP",ifelse(grepl("17", imp.all.melted$variable), "17 DAP",ifelse(grepl("21", imp.all.melted$variable), "21 DAP", "na"))))

imp.all.melted$exp <- ifelse(imp.all.melted$maternal_preference >= .8, "possible MEG", ifelse(imp.all.melted$maternal_preference < .4, "possible PEG", "biparental" ))

library(readxl)
library(tidyverse)
library(ggplot2)
library(dplyr)
library(tidyr)
library(ggExtra)

gene_list<-data.frame(unique(imp.all.melted$feature))
my_gene<- function(gene){                         #function will take in a value(gene)
  gene_list<-data.frame(unique(imp.all.melted$feature))
  gene_data <- imp.all.melted[imp.all.melted$feature == gene, ]#search through the original data and eliminate all the rows that do not mention the gene. left with gene data
  gene_exp <- imp.rpm.melted[imp.rpm.melted$feature == gene,]
  gene_mat_pref <- gene_data %>% ggplot(aes(x=Timepoint , y= maternal_preference, group =1, color = exp))+
    geom_point()+ggtitle(gene)+ geom_line()+ylab("Maternal Preference" )+xlab("Days After Pollination")+ scale_color_manual("Predicted expression",values = c("possible MEG" = "maroon", "possible PEG" = "navy", "biparental" = "darkgrey")) + ylim(0,1) +  theme_classic()
  gene_expression <- gene_exp %>% ggplot(aes(x=Timepoint, y=value, color=group2)) + geom_point() +ylab("Normalized Expression in RPM") + xlab("Days After Pollination")+ theme_classic()
  grid.arrange(gene_mat_pref,gene_expression, ncol = 1)
}

```

```{r}
#Make alluvial plot of results
library(tidyr)
library(ggalluvial)
library(stringr)

imp.rpm.all$pref_11 <- ifelse(imp.rpm.all$genome =="B73", imp.rpm.all$A_mean11/(imp.rpm.all$B_mean11 + imp.rpm.all$A_mean11), imp.rpm.all$B_mean11/(imp.rpm.all$B_mean11 + imp.rpm.all$A_mean11))
imp.rpm.all$pref_14 <- ifelse(imp.rpm.all$genome =="B73", imp.rpm.all$A_mean14/(imp.rpm.all$B_mean14 + imp.rpm.all$A_mean14), imp.rpm.all$B_mean14/(imp.rpm.all$B_mean14 + imp.rpm.all$A_mean14))
imp.rpm.all$pref_17 <- ifelse(imp.rpm.all$genome =="B73", imp.rpm.all$A_mean17/(imp.rpm.all$B_mean17 + imp.rpm.all$A_mean17), imp.rpm.all$B_mean17/(imp.rpm.all$B_mean17 + imp.rpm.all$A_mean17))
imp.rpm.all$pref_21 <- ifelse(imp.rpm.all$genome =="B73", imp.rpm.all$A_mean21/(imp.rpm.all$B_mean21 + imp.rpm.all$A_mean21), imp.rpm.all$B_mean21/(imp.rpm.all$B_mean21 + imp.rpm.all$A_mean21))

imp.rpm.all$exp11 <- ifelse(imp.rpm.all$pref_11 >= .9 , "maternal", ifelse(imp.rpm.all$pref_11 <= .3, "paternal",ifelse(imp.rpm.all$pref_11 > .3 & imp.rpm.all$pref_11 <.4, "bordering paternal", ifelse(imp.rpm.all$pref_11 >.8 & imp.rpm.all$pref_11 < .9, "bordering maternal", "biparental"))))

imp.rpm.all$exp14 <- ifelse(imp.rpm.all$pref_14 >= .9 , "maternal", ifelse(imp.rpm.all$pref_14 <= .3, "paternal",ifelse(imp.rpm.all$pref_14 > .3 & imp.rpm.all$pref_14 <.4, "bordering paternal", ifelse(imp.rpm.all$pref_14 >.8 & imp.rpm.all$pref_14 < .9, "bordering maternal", "biparental"))))

imp.rpm.all$exp17 <- ifelse(imp.rpm.all$pref_17 >= .9 , "maternal", ifelse(imp.rpm.all$pref_17 <= .3, "paternal",ifelse(imp.rpm.all$pref_17 > .3 & imp.rpm.all$pref_17 <.4, "bordering paternal", ifelse(imp.rpm.all$pref_17 >.8 & imp.rpm.all$pref_17 < .9, "bordering maternal", "biparental"))))

imp.rpm.all$exp21 <- ifelse(imp.rpm.all$pref_21 >= .9 , "maternal", ifelse(imp.rpm.all$pref_21 <= .3, "paternal",ifelse(imp.rpm.all$pref_21 > .3 & imp.rpm.all$pref_21 <.4, "bordering paternal", ifelse(imp.rpm.all$pref_21 >.8 & imp.rpm.all$pref_21 < .9, "bordering maternal", "biparental"))))


imp.rpm.justcalls <- imp.rpm.all[,c(1,19:22)]
imp.rpm.onlyimps <- subset(imp.rpm.justcalls, imp.rpm.justcalls$exp11 != "biparental" | imp.rpm.justcalls$exp17 != "biparental" |imp.rpm.justcalls$exp14 != "biparental" | imp.rpm.justcalls$exp21 != "biparental")

alluvial_DE_gene <- ggplot(data = imp.rpm.onlyimps, aes(axis1 = exp11, axis3 = exp14, axis2 = exp17, axis4 = exp21)) + geom_alluvium(aes(fill = exp14),curve_type = "cubic") + geom_stratum() + geom_text(stat = "stratum",aes(label = after_stat(stratum))) + coord_flip() + ggtitle("imprints over time") + theme_set(theme_void() + theme(legend.position = "bottom"))

alluvial_DE_gene

imp.rpm.onlyimps$pattern <- paste(imp.rpm.onlyimps$exp11, "-", imp.rpm.onlyimps$exp14, "-",imp.rpm.onlyimps$exp17,"-",imp.rpm.onlyimps$exp21)

imprinting_patterns <- imp.rpm.onlyimps %>%
  group_by(pattern) %>%
  summarise(n=n())%>%
  mutate(Freq=n/sum(n))

mat_only_pattern <- subset(imp.rpm.onlyimps, imp.rpm.onlyimps$pattern == "maternal - maternal - maternal - maternal")

B73_mat_only <- subset(mat_only_pattern$feature, grepl("^Zm00001eb", mat_only_pattern$feature))
W22_mat_only <- subset(mat_only_pattern$feature, grepl("^Zm00004", mat_only_pattern$feature))

```
```{r}
#Identify any overall patterns for RER across time using kmeans
# Install and load necessary packages
if (!requireNamespace("pheatmap", quietly = TRUE)) {
  install.packages("pheatmap")
}
if (!requireNamespace("factoextra", quietly = TRUE)) {
  install.packages("factoextra")
}
if (!requireNamespace("plotly", quietly = TRUE)) {
  install.packages("plotly")
}
if (!requireNamespace("NbClust", quietly = TRUE)) {
  install.packages("NbClust")
}


library(NbClust)
library(tidyverse)
library(lubridate)
library(pwt9)
library(purrr)
library(stats)
library(pheatmap)
library(factoextra)
library(plotly)
library(factoextra)

```


```{r}
head(imp.mat.all)

#Remove unneccesary columns 
kmeans_RER <- imp.mat.all[,c(1,2,4,6,8)]

#Remove NAs
kmeans_RER[is.na(kmeans_RER)] <- 0
```

```{r}
give.n <- function(x){
   return(c(y = mean(x), label = length(x)))
}
```


```{r}
#Calculate imprinting for each time point and plot 

##Remove NA's from dfs
imp.11DAP.rpm2 <- subset(imp.11DAP.rpm, imp.11DAP.rpm$maternal_preference != "NaN")
imp.14DAP.rpm2 <- subset(imp.14DAP.rpm, imp.14DAP.rpm$maternal_preference != "NaN")
imp.17DAP.rpm2 <- subset(imp.17DAP.rpm, imp.17DAP.rpm$maternal_preference != "NaN")
imp.21DAP.rpm2 <- subset(imp.21DAP.rpm, imp.21DAP.rpm$maternal_preference != "NaN")

##Remove genes with very low expression
imp.11.filter <- subset(imp.11DAP.rpm2,imp.11DAP.rpm2$A_mean >= 1 | imp.11DAP.rpm2$B_mean >= 1)
imp.14.filter <- subset(imp.14DAP.rpm2,imp.14DAP.rpm2$A_mean >= 1 | imp.14DAP.rpm2$B_mean >= 1)
imp.17.filter <- subset(imp.17DAP.rpm2,imp.17DAP.rpm2$A_mean >= 1 | imp.17DAP.rpm2$B_mean >= 1)
imp.21.filter <- subset(imp.21DAP.rpm2,imp.21DAP.rpm2$A_mean >= 1 | imp.21DAP.rpm2$B_mean >= 1)

elevenDAP_plot <- ggplot(imp.11.filter,aes(x=genome,y=maternal_preference,fill=genome))+ geom_abline(slope=0, intercept=.33) + geom_abline(slope=0, intercept=.66) + geom_violin() + theme_classic() + theme(panel.grid.major = element_blank(),panel.grid.minor = element_blank()) + facet_wrap(.~contrast,scales="free_x")  + stat_summary(fun.data = give.n, geom = "text") + scale_fill_manual(values = maize_pal("HighlandMAGIC"))
fourteenDAP_plot <- ggplot(imp.14.filter,aes(x=genome,y=maternal_preference,fill=genome))+ geom_abline(slope=0, intercept=.33) + geom_abline(slope=0, intercept=.66) + geom_violin() + theme_classic() + theme(panel.grid.major = element_blank(),panel.grid.minor = element_blank()) + facet_wrap(.~contrast,scales="free_x")  + stat_summary(fun.data = give.n, geom = "text") + scale_fill_manual(values = maize_pal("HighlandMAGIC"))
seventeenDAP_plot <- ggplot(imp.17.filter,aes(x=genome,y=maternal_preference,fill=genome))+ geom_abline(slope=0, intercept=.33) + geom_abline(slope=0, intercept=.66) + geom_violin() + theme_classic() + theme(panel.grid.major = element_blank(),panel.grid.minor = element_blank()) + facet_wrap(.~contrast,scales="free_x")  + stat_summary(fun.data = give.n, geom = "text") + scale_fill_manual(values = maize_pal("HighlandMAGIC"))
twentyoneDAP_plot <- ggplot(imp.21.filter,aes(x=genome,y=maternal_preference,fill=genome))+ geom_abline(slope=0, intercept=.33) + geom_abline(slope=0, intercept=.66) + geom_violin() + theme_classic() + theme(panel.grid.major = element_blank(),panel.grid.minor = element_blank()) + facet_wrap(.~contrast,scales="free_x")  + stat_summary(fun.data = give.n, geom = "text") + scale_fill_manual(values = maize_pal("HighlandMAGIC"))

grid.arrange(elevenDAP_plot, fourteenDAP_plot,seventeenDAP_plot,twentyoneDAP_plot)
```


Use DEseq to call significance over a log2FC cutoff of 1 (2-fold, which is the expected ratio) do this for each df
```{r}
imp.all.de.11DAP <- counts[,1:6]
rownames(imp.all.de.11DAP) <- row.names(counts)
#imp.all.de.keep.11DAP <- subset(imp.all.de.11DAP,rowSums(imp.all.de.11DAP >0) >= 3) 
imp.all.de.keep.11DAP <- subset(imp.all.de.11DAP,rowSums(imp.all.de.11DAP) >= 10) 
imp.all.de.keep.11DAP <- subset(imp.all.de.11DAP,rowMeans(counts.rpm2[,1:3]) >= 0.5 | rowMeans(counts.rpm2[,4:6]) >= 0.5) 

sample.info.11DAP <- as.data.frame(cbind(paste(names(imp.all.de.keep.11DAP)),c("BW","BW","BW","WB","WB","WB")))
rownames(sample.info.11DAP) <- sample.info.11DAP[,1]
sample_list.11DAP <- sample.info.11DAP[,2]
names(sample.info.11DAP) <- c("sample.info.11DAP","sample_list.11DAP")

dds.11DAP <- DESeqDataSetFromMatrix(countData = imp.all.de.keep.11DAP,colData = sample.info.11DAP, design = ~ sample_list.11DAP)
dds.11DAP <- DESeq(dds.11DAP)
res.11DAP <- results(dds.11DAP, lfcThreshold=1, altHypothesis = "greaterAbs", cooksCutoff = F, independentFiltering = F)
drawLines <- function() abline(h=c(-1,1),col="dodgerblue",lwd=2)
#par(mfrow=c(2,2),mar=c(2,2,1,1))
ylim <- c(-2.5,2.5)
plotMA(res.11DAP, ylim=ylim); drawLines()

rld <- rlog(dds.11DAP, blind=FALSE)
a <- plotPCA(rld,intgroup="sample_list.11DAP",ntop = 10000) + theme_classic()
a
```

Plot results with respect to RER
```{r}
out.11DAP <- data.frame(res.11DAP)
plot.out.11DAP <- merge(imp.11DAP.rpm2,out.11DAP,by.x="feature",by.y="row.names",all=F)
plot.out.11DAP$sig <- ifelse(plot.out.11DAP$padj < 0.01, "TRUE", "FALSE")
plot.out2.11DAP <- subset(plot.out.11DAP, plot.out.11DAP$contrast %in% c("11DAP"))
plot.out2.11DAP$category <- ifelse(plot.out2.11DAP$padj < 0.05 & plot.out2.11DAP$log2FoldChange < -1 & (plot.out2.11DAP$maternal_preference > .9 | plot.out2.11DAP$maternal_preference < .3), "up",ifelse(plot.out2.11DAP$padj < 0.05 & plot.out2.11DAP$log2FoldChange > 1 & (plot.out2.11DAP$maternal_preference > .9 | plot.out2.11DAP$maternal_preference < .3),"down","notDE"))

plot.out2.11DAP$imprint <- ifelse(plot.out2.11DAP$category =="up" & plot.out2.11DAP$genome == "B73","MEG",
                               ifelse(plot.out2.11DAP$category =="down" & plot.out2.11DAP$genome == "B73","PEG",
                                      ifelse(plot.out2.11DAP$category =="down" & plot.out2.11DAP$genome == "W22","MEG",
                                             ifelse(plot.out2.11DAP$category =="up" & plot.out2.11DAP$genome == "W22","PEG", "no.imprint"))))

tmp <- subset(plot.out2.11DAP, is.na(plot.out2.11DAP$imprint))

e <- ggplot(plot.out2.11DAP,aes(x=imprint,y=maternal_preference,fill=imprint)) + geom_violin() + theme_classic() + theme(panel.grid.major = element_blank(),panel.grid.minor = element_blank()) + facet_grid(genome~.) + geom_abline(slope=0, intercept=.33) + geom_abline(slope=0, intercept=.66) + stat_summary(fun.data = give.n, geom = "text") + scale_fill_manual(values = maize_pal("HighlandMAGIC"))+ ggtitle("11 Days after pollination")

e
```




```{r}
imp.all.de.14DAP <- counts[,7:12]
rownames(imp.all.de.14DAP) <- row.names(counts)
#imp.all.de.keep.14DAP <- subset(imp.all.de.14DAP,rowSums(imp.all.de.14DAP >0) >= 3) 
imp.all.de.keep.14DAP <- subset(imp.all.de.14DAP,rowSums(imp.all.de.14DAP) >= 10) 
imp.all.de.keep.14DAP <- subset(imp.all.de.14DAP,rowMeans(counts.rpm2[,7:9]) >= 0.5 | rowMeans(counts.rpm2[,10:12]) >= 0.5) 

sample.info.14DAP <- as.data.frame(cbind(paste(names(imp.all.de.keep.14DAP)),c("BW","BW","BW","WB","WB","WB")))
rownames(sample.info.14DAP) <- sample.info.14DAP[,1]
sample_list.14DAP <- sample.info.14DAP[,2]
names(sample.info.14DAP) <- c("sample.info.14DAP","sample_list.14DAP")

dds.14DAP <- DESeqDataSetFromMatrix(countData = imp.all.de.keep.14DAP,colData = sample.info.14DAP, design = ~ sample_list.14DAP)
dds.14DAP <- DESeq(dds.14DAP)
res.14DAP <- results(dds.14DAP, lfcThreshold=1, altHypothesis = "greaterAbs")
drawLines <- function() abline(h=c(-1,1),col="dodgerblue",lwd=2)
#par(mfrow=c(2,2),mar=c(2,2,1,1))
ylim <- c(-2.5,2.5)
plotMA(res.14DAP, ylim=ylim); drawLines()

rld <- rlog(dds.14DAP, blind=FALSE)
a <- plotPCA(rld,intgroup="sample_list.14DAP",ntop = 10000) +theme_classic()
```

Plot results with respect to RER
```{r}
out.14DAP <- data.frame(res.14DAP)
plot.out.14DAP <- merge(imp.14DAP.rpm2,out.14DAP,by.x="feature",by.y="row.names",all=F)
plot.out.14DAP$sig <- ifelse(plot.out.14DAP$padj < 0.01, "TRUE", "FALSE")
plot.out2.14DAP <- subset(plot.out.14DAP,!is.na(plot.out.14DAP$sig) & plot.out.14DAP$contrast %in% c("14DAP"))
plot.out2.14DAP$category <- ifelse(plot.out2.14DAP$padj < 0.05 & plot.out2.14DAP$log2FoldChange < -1 & (plot.out2.14DAP$maternal_preference > .9 | plot.out2.14DAP$maternal_preference < .3), "up",ifelse(plot.out2.14DAP$padj < 0.05 & plot.out2.14DAP$log2FoldChange > 1 & (plot.out2.14DAP$maternal_preference > .9 | plot.out2.14DAP$maternal_preference < .3),"down","notDE"))

plot.out2.14DAP$imprint <- ifelse(plot.out2.14DAP$category =="up" & plot.out2.14DAP$genome == "B73","MEG",
                               ifelse(plot.out2.14DAP$category =="down" & plot.out2.14DAP$genome == "B73","PEG",
                                      ifelse(plot.out2.14DAP$category =="down" & plot.out2.14DAP$genome == "W22","MEG",
                                             ifelse(plot.out2.14DAP$category =="up" & plot.out2.14DAP$genome == "W22","PEG","no.imprint"))))

f <- ggplot(plot.out2.14DAP,aes(x=imprint,y=maternal_preference,fill=imprint)) + geom_violin() + theme_classic() + theme(panel.grid.major = element_blank(),panel.grid.minor = element_blank()) + facet_grid(genome~.) + geom_abline(slope=0, intercept=.33) + geom_abline(slope=0, intercept=.66) + stat_summary(fun.data = give.n, geom = "text") + scale_fill_manual(values = maize_pal("HighlandMAGIC")) + ggtitle("14 Days after pollination")

f
```



Use DEseq to call significance over a log2FC cutoff of 1 (2-fold, which is the expected ratio) do this for each df
```{r}
imp.all.de.17DAP <- counts[,13:18]
rownames(imp.all.de.17DAP) <- row.names(counts)
#imp.all.de.keep.17DAP <- subset(imp.all.de.17DAP,rowSums(imp.all.de.17DAP >0) >= 3) 
imp.all.de.keep.17DAP <- subset(imp.all.de.17DAP,rowSums(imp.all.de.17DAP) >= 10) 
imp.all.de.keep.17DAP <- subset(imp.all.de.17DAP,rowMeans(counts.rpm2[,13:15]) >= 0.5 | rowMeans(counts.rpm2[,16:18]) >= 0.5) 

sample.info.17DAP <- as.data.frame(cbind(paste(names(imp.all.de.keep.17DAP)),c("BW","BW","BW","WB","WB","WB")))
rownames(sample.info.17DAP) <- sample.info.17DAP[,1]
sample_list.17DAP <- sample.info.17DAP[,2]
names(sample.info.17DAP) <- c("sample.info.17DAP","sample_list.17DAP")

dds.17DAP <- DESeqDataSetFromMatrix(countData = imp.all.de.keep.17DAP,colData = sample.info.17DAP, design = ~ sample_list.17DAP)
dds.17DAP <- DESeq(dds.17DAP)
res.17DAP <- results(dds.17DAP, lfcThreshold=1, altHypothesis = "greaterAbs")
drawLines <- function() abline(h=c(-1,1),col="dodgerblue",lwd=2)
#par(mfrow=c(2,2),mar=c(2,2,1,1))
ylim <- c(-2.5,2.5)
plotMA(res.17DAP, ylim=ylim); drawLines()

rld <- rlog(dds.17DAP, blind=FALSE)
a <- plotPCA(rld,intgroup="sample_list.17DAP",ntop = 10000) + theme_classic()

```

Plot results with respect to RER
```{r}
out.17DAP <- data.frame(res.17DAP)
plot.out.17DAP <- merge(imp.17DAP.rpm2,out.17DAP,by.x="feature",by.y="row.names",all=F)
plot.out.17DAP$sig <- ifelse(plot.out.17DAP$padj < 0.01, "TRUE", "FALSE")
plot.out2.17DAP <- subset(plot.out.17DAP,!is.na(plot.out.17DAP$sig) & plot.out.17DAP$contrast %in% c("17DAP"))
plot.out2.17DAP$category <- ifelse(plot.out2.17DAP$padj < 0.05 & plot.out2.17DAP$log2FoldChange < -1 & (plot.out2.17DAP$maternal_preference > .9 | plot.out2.17DAP$maternal_preference < .3), "up",ifelse(plot.out2.17DAP$padj < 0.05 & plot.out2.17DAP$log2FoldChange > 1 & (plot.out2.17DAP$maternal_preference > .9 | plot.out2.17DAP$maternal_preference < .3),"down","notDE"))

plot.out2.17DAP$imprint <- ifelse(plot.out2.17DAP$category =="up" & plot.out2.17DAP$genome == "B73","MEG",
                               ifelse(plot.out2.17DAP$category =="down" & plot.out2.17DAP$genome == "B73","PEG",
                                      ifelse(plot.out2.17DAP$category =="down" & plot.out2.17DAP$genome == "W22","MEG",
                                             ifelse(plot.out2.17DAP$category =="up" & plot.out2.17DAP$genome == "W22","PEG","no.imprint"))))

g <- ggplot(plot.out2.17DAP,aes(x=imprint,y=maternal_preference,fill=imprint)) + geom_violin() + theme_classic() + theme(panel.grid.major = element_blank(),panel.grid.minor = element_blank()) + facet_grid(genome~.) + geom_abline(slope=0, intercept=.33) + geom_abline(slope=0, intercept=.66) + stat_summary(fun.data = give.n, geom = "text") + scale_fill_manual(values = maize_pal("HighlandMAGIC"))+ ggtitle("17 Days after pollination")

g
```




Use DEseq to call significance over a log2FC cutoff of 1 (2-fold, which is the expected ratio)
```{r}
imp.all.de.21DAP <- counts[,19:23]
rownames(imp.all.de.21DAP) <- row.names(counts)
imp.all.de.keep.21DAP <- subset(imp.all.de.21DAP,rowSums(imp.all.de.21DAP) >= 3) 
imp.all.de.keep.21DAP <- subset(imp.all.de.21DAP,rowSums(imp.all.de.21DAP) >= 10) 
imp.all.de.keep.21DAP <- subset(imp.all.de.21DAP,rowMeans(counts.rpm2[,19:21]) >= 0.5 | rowMeans(counts.rpm2[,22:23]) >= 0.5) 

sample.info.21DAP <- as.data.frame(cbind(paste(names(imp.all.de.keep.21DAP)),c("BW","BW","BW","WB","WB")))
rownames(sample.info.21DAP) <- sample.info.21DAP[,1]
sample_list.21DAP <- sample.info.21DAP[,2]
names(sample.info.21DAP) <- c("sample.info.21DAP","sample_list.21DAP")

dds.21DAP <- DESeqDataSetFromMatrix(countData = imp.all.de.keep.21DAP,colData = sample.info.21DAP, design = ~ sample_list.21DAP)
dds.21DAP <- DESeq(dds.21DAP)
res.21DAP <- results(dds.21DAP, lfcThreshold=1, altHypothesis = "greaterAbs")
drawLines <- function() abline(h=c(-1,1),col="dodgerblue",lwd=2)
#par(mfrow=c(2,2),mar=c(2,2,1,1))
ylim <- c(-2.5,2.5)
plotMA(res.21DAP, ylim=ylim); drawLines()

rld <- rlog(dds.21DAP, blind=FALSE)
a <- plotPCA(rld,intgroup="sample_list.21DAP",ntop = 10000)
a + theme_classic()

```

Plot results with respect to RER
```{r}
out.21DAP <- data.frame(res.21DAP)
plot.out.21DAP <- merge(imp.21DAP.rpm2,out.21DAP,by.x="feature",by.y="row.names",all=F)
plot.out.21DAP$sig <- ifelse(plot.out.21DAP$padj < 0.01, "TRUE", "FALSE")
plot.out2.21DAP <- subset(plot.out.21DAP,!is.na(plot.out.21DAP$sig) & plot.out.21DAP$contrast %in% c("21DAP"))
plot.out2.21DAP$category <- ifelse(plot.out2.21DAP$padj < 0.05 & plot.out2.21DAP$log2FoldChange < -1 & (plot.out2.21DAP$maternal_preference > .9 | plot.out2.21DAP$maternal_preference < .3), "up",ifelse(plot.out2.21DAP$padj < 0.05 & plot.out2.21DAP$log2FoldChange > 1 & (plot.out2.21DAP$maternal_preference > .9 | plot.out2.21DAP$maternal_preference < .3),"down","notDE"))

plot.out2.21DAP$imprint <- ifelse(plot.out2.21DAP$category =="up" & plot.out2.21DAP$genome == "B73","MEG",
                               ifelse(plot.out2.21DAP$category =="down" & plot.out2.21DAP$genome == "B73","PEG",
                                      ifelse(plot.out2.21DAP$category =="down" & plot.out2.21DAP$genome == "W22","MEG",
                                             ifelse(plot.out2.21DAP$category =="up" & plot.out2.21DAP$genome == "W22","PEG","no.imprint"))))

h <- ggplot(plot.out2.21DAP,aes(x=imprint,y=maternal_preference,fill=imprint)) + geom_violin() + theme_classic() + theme(panel.grid.major = element_blank(),panel.grid.minor = element_blank()) + facet_grid(genome~.) + geom_abline(slope=0, intercept=.33) + geom_abline(slope=0, intercept=.66) + stat_summary(fun.data = give.n, geom = "text") + scale_fill_manual(values = maize_pal("HighlandMAGIC"))+ ggtitle("21 Days after pollination")

h

grid.arrange(e,f,g,h, nrow = 2)
```


```{r}
summary.11 <- data.frame(plot.out2.11DAP %>% group_by(genome,imprint,contrast) %>% summarize(count = n()))
summary.14 <- data.frame(plot.out2.14DAP %>% group_by(genome,imprint,contrast) %>% summarize(count = n()))
summary.21 <- data.frame(plot.out2.21DAP %>% group_by(genome,imprint,contrast) %>% summarize(count = n()))
summary.17 <- data.frame(plot.out2.17DAP %>% group_by(genome,imprint,contrast) %>% summarize(count = n()))

summary.combined <- rbind(summary.11,summary.14,summary.17,summary.21)
summary.combined

summary.combined2 <- subset(summary.combined,summary.combined$imprint %in% c("MEG","PEG"))
summary.combined2[summary.combined2$imprint == "PEG","count"] <- -summary.combined2[summary.combined2$imprint == "PEG","count"]
summary.combined2$label <- paste(summary.combined2$genome,summary.combined2$contrast,sep="_")
summary.combined2$label <- factor(summary.combined2$label, levels = c("B73_11DAP","W22_11DAP","B73_14DAP","W22_14DAP","B73_17DAP","W22_17DAP","B73_21DAP","W22_21DAP"))

ggplot() + geom_bar(aes(x=label,y=count,fill=imprint),data=summary.combined2,stat="identity",position = "identity") + theme_classic()  +  theme(axis.text.x = element_text(angle=90, hjust=1)) + scale_fill_manual(values=c("magenta2","navy")) 

```

```{r}
#Next I want to know how many genes are imprinted at all timepoints vs ones that change
 # First subset imprinted genes from each timepoint
Imprints_11DAP <- subset(plot.out2.11DAP[,c(1,22)], plot.out2.11DAP$imprint != "no.imprint")
Imprints_14DAP <- subset(plot.out2.14DAP[,c(1,22)], plot.out2.14DAP$imprint != "no.imprint")
Imprints_17DAP <- subset(plot.out2.17DAP[,c(1,22)], plot.out2.17DAP$imprint != "no.imprint")
Imprints_21DAP <- subset(plot.out2.21DAP[,c(1,21)], plot.out2.21DAP$imprint != "no.imprint")
  #Next bind those dataframes together so we can assess imprinting across time
Imprints_11and14 <- full_join(Imprints_11DAP, Imprints_14DAP, by = "feature")
Imprints_11_14_17 <- full_join(Imprints_11and14, Imprints_17DAP, by = "feature")
Imprints_all <- full_join(Imprints_11_14_17, Imprints_21DAP, by = "feature")
  #Change NAs in the df to no imprint 
Imprints_all[is.na(Imprints_all)] <- "No Imprint"
  #Rename columns to make the df easily understood
colnames(Imprints_all) <- c("feature", "imprint.11", "imprint.14", "imprint.17", "imprint.21")
  #Create counts of imprinting for each gene
Imprinting_counts <- Imprints_all %>% group_by(imprint.11,imprint.14,imprint.17,imprint.21) %>% count()

#Create venn diagram of overlaps 
library(ggVennDiagram)
#Subset Imprinting dfs so they are MEG or PEG
MEG_11 <- subset(Imprints_11DAP, Imprints_11DAP$imprint == "MEG")
MEG_14 <- subset(Imprints_14DAP, Imprints_14DAP$imprint == "MEG")
MEG_17 <- subset(Imprints_17DAP, Imprints_17DAP$imprint == "MEG")
MEG_21 <- subset(Imprints_21DAP, Imprints_21DAP$imprint == "MEG")

PEG_11 <- subset(Imprints_11DAP, Imprints_11DAP$imprint == "PEG")
PEG_14 <- subset(Imprints_14DAP, Imprints_14DAP$imprint == "PEG")
PEG_17 <- subset(Imprints_17DAP, Imprints_17DAP$imprint == "PEG")
PEG_21 <- subset(Imprints_21DAP, Imprints_21DAP$imprint == "PEG")


#Same thing, but look at feature type too
MEG_intersection <- list(A = MEG_11$feature, B = MEG_14$feature, C = MEG_17$feature, D = MEG_21$feature)
PEG_intersection <- list(A = PEG_11$feature, B = PEG_14$feature, C = PEG_17$feature, D = PEG_21$feature)
ggVennDiagram(MEG_intersection) + scale_fill_gradient(low = "pink", high = "maroon") + ggtitle("matgenes")
ggVennDiagram(PEG_intersection) + scale_fill_gradient(low = "skyblue", high = "navy") + ggtitle("patgenes")

#write files of consistent MEGs and PEGs, legitimately just the genes for comparison across lines. Maybe highly imprinted genes are more conserved across lines. 

write.table(Imprints_all, "//las-dfs-01.las.iastate.edu/lss/research/sna-lab/kmh/projects/timepoint/Imprints_all_timepoints.txt", quote = F, row.names = F, col.names = T)
```

```{r}
#Use my_gene function to look at all genes in the BC group of MEGs to determine if there is a pattern of expression happening. 
BC_MEG <- subset(Imprints_all, Imprints_all$imprint.11 != "MEG" & Imprints_all$imprint.14 == "MEG" & Imprints_all$imprint.17 == "MEG"& Imprints_all$imprint.21 != "MEG" )

bc_meg_list <- BC_MEG$feature

#No major pattern I'm seeing with these 10 other than stringent cutoffs, most never have the paternal allele turn on really
    #Look at A subset of MEGs, and D subset of PEGs

A_MEGs <- subset(Imprints_all, Imprints_all$imprint.11 == "MEG" & Imprints_all$imprint.14 != "MEG" & Imprints_all$imprint.17 != "MEG"& Imprints_all$imprint.21 != "MEG" )

D_PEGs <- subset(Imprints_all, Imprints_all$imprint.11 != "PEG" & Imprints_all$imprint.14 != "PEG" & Imprints_all$imprint.17 != "PEG"& Imprints_all$imprint.21 == "PEG" )

a_meg_list <- A_MEGs$feature
d_peg_list <- D_PEGs$feature

#No discernable pattern for all of these, and need a different metric to show changes in parental expression.
    #Normalize all maternal and paternal expression so it has a range from zero to one
imp.11DAP.rpm$A_norm <- (imp.11DAP.rpm$A_mean)/rowMaxs(as.matrix(imp.11DAP.rpm[,c("A_mean", "B_mean")]))
imp.14DAP.rpm$A_norm <- (imp.14DAP.rpm$A_mean)/rowMaxs(as.matrix(imp.14DAP.rpm[,c("A_mean", "B_mean")]))
imp.17DAP.rpm$A_norm <- (imp.17DAP.rpm$A_mean)/rowMaxs(as.matrix(imp.17DAP.rpm[,c("A_mean", "B_mean")]))
imp.21DAP.rpm$A_norm <- (imp.21DAP.rpm$A_mean)/rowMaxs(as.matrix(imp.21DAP.rpm[,c("A_mean", "B_mean")]))
imp.11DAP.rpm$B_norm <- (imp.11DAP.rpm$B_mean)/rowMaxs(as.matrix(imp.11DAP.rpm[,c("A_mean", "B_mean")]))
imp.14DAP.rpm$B_norm <- (imp.14DAP.rpm$B_mean)/rowMaxs(as.matrix(imp.14DAP.rpm[,c("A_mean", "B_mean")]))
imp.17DAP.rpm$B_norm <- (imp.17DAP.rpm$B_mean)/rowMaxs(as.matrix(imp.17DAP.rpm[,c("A_mean", "B_mean")]))
imp.21DAP.rpm$B_norm <- (imp.21DAP.rpm$B_mean)/rowMaxs(as.matrix(imp.21DAP.rpm[,c("A_mean", "B_mean")]))

#Get relevant info into one df 
mat_dom <- imp.11DAP.rpm[,c("feature", "A_norm","B_norm")]
mat_dom2 <- left_join(mat_dom, imp.14DAP.rpm[,c("feature", "A_norm","B_norm")], by = "feature")
mat_dom3 <- left_join(mat_dom2, imp.17DAP.rpm[,c("feature", "A_norm","B_norm")], by = "feature")
mat_dom_all <-  left_join(mat_dom3, imp.21DAP.rpm[,c("feature", "A_norm","B_norm")], by = "feature")
colnames(mat_dom_all) <- c("feature","11_A","11_B","14_A","14_B","17_A","17_B","21_A","21_B")
mat_dom_all$type <- ifelse(grepl("Zm00004b",mat_dom_all$feature), "B", "A")

#Melt maternal dominance df 
mat_dom_melted <- melt(mat_dom_all)
mat_dom_melted$allele <- ifelse(grepl("A",mat_dom_melted$variable) & mat_dom_melted$type == "A", "maternal", ifelse(grepl("B",mat_dom_melted$variable) & mat_dom_melted$type == "B", "maternal", "paternal"))
#use substring to get timepoint
mat_dom_melted$time <- substr(mat_dom_melted$variable,1,2)
#Add a category for maternal and paternal
mat_dom_melted$feature_exp <- paste(mat_dom_melted$feature,"-", mat_dom_melted$allele)


#Take the my gene function parts and try to get everything on one plot
gene_patterns <- function(df,title){ gene_list<-data.frame(unique(imp.all.melted$feature))
  gene_data <- imp.all.melted[imp.all.melted$feature %in% df, ]#search through the original data and eliminate all the rows that do not mention the gene. left with gene data
  gene_exp <- mat_dom_melted[imp.rpm.melted$feature %in% df,]
  gene_mat_pref <- gene_data %>% ggplot(aes(x=Timepoint , y= maternal_preference, group =feature, color = exp))+
    geom_point()+ggtitle(title)+ geom_line()+ylab("Maternal Preference" )+xlab("Days After Pollination")+ scale_color_manual("Predicted expression",values = c("possible MEG" = "maroon", "possible PEG" = "navy", "biparental" = "darkgrey")) + ylim(0,1) +  theme_classic()
  gene_expression <- gene_exp %>% ggplot(aes(x=time, y=value, group=feature_exp, color = allele)) + geom_point() +ylab("Normalized allele expression") + xlab("Days After Pollination")+ theme_classic() +geom_line()
  grid.arrange(gene_mat_pref,gene_expression, ncol = 1)
}


gene_patterns(a_meg_list, "MEGs at only 11DAP")
gene_patterns(bc_meg_list, "MEGs at 14 and 17DAP only")
gene_patterns(d_peg_list, "PEGs at only 21DAP")

#Now that I've tested this, I need to look at MEGs and PEGs expressed at only one timepoint 
B_MEGs <- subset(Imprints_all, Imprints_all$imprint.11 != "MEG" & Imprints_all$imprint.14 == "MEG" & Imprints_all$imprint.17 != "MEG"& Imprints_all$imprint.21 != "MEG" )
C_MEGs <- subset(Imprints_all, Imprints_all$imprint.11 != "MEG" & Imprints_all$imprint.14 != "MEG" & Imprints_all$imprint.17 == "MEG"& Imprints_all$imprint.21 != "MEG" )
D_MEGs <- subset(Imprints_all, Imprints_all$imprint.11 != "MEG" & Imprints_all$imprint.14 != "MEG" & Imprints_all$imprint.17 != "MEG"& Imprints_all$imprint.21 == "MEG" )

b_meg_list <- B_MEGs$feature
c_meg_list <- C_MEGs$feature
d_meg_list <- D_MEGs$feature

A_PEGs <- subset(Imprints_all, Imprints_all$imprint.11 == "PEG" & Imprints_all$imprint.14 != "PEG" & Imprints_all$imprint.17 != "PEG"& Imprints_all$imprint.21 != "PEG" )
B_PEGs <- subset(Imprints_all, Imprints_all$imprint.11 != "PEG" & Imprints_all$imprint.14 == "PEG" & Imprints_all$imprint.17 != "PEG"& Imprints_all$imprint.21 != "PEG" )
C_PEGs <- subset(Imprints_all, Imprints_all$imprint.11 != "PEG" & Imprints_all$imprint.14 != "PEG" & Imprints_all$imprint.17 == "PEG"& Imprints_all$imprint.21 != "PEG" )
D_PEGs <- subset(Imprints_all, Imprints_all$imprint.11 != "PEG" & Imprints_all$imprint.14 != "PEG" & Imprints_all$imprint.17 != "PEG"& Imprints_all$imprint.21 == "PEG" )

a_peg_list <- A_PEGs$feature
b_peg_list <- B_PEGs$feature
c_peg_list <- C_PEGs$feature
d_peg_list <- D_PEGs$feature

gene_patterns(d_meg_list,"MEGs at only 21DAP")
gene_patterns(b_meg_list,"MEGs at only 14DAP")
gene_patterns(c_meg_list,"MEGs at only 17DAP")

gene_patterns(a_peg_list, "PEGs at only 11DAP")
gene_patterns(b_peg_list,"PEGs at only 14DAP")
gene_patterns(c_peg_list,"PEGs at only 17DAP")

#Hell, lets look at the consistent ones to see if its all just easy to understand
all_MEGs <- subset(Imprints_all, Imprints_all$imprint.11 == "MEG" & Imprints_all$imprint.14 == "MEG" & Imprints_all$imprint.17 == "MEG"& Imprints_all$imprint.21 == "MEG" )

all_PEGs <- subset(Imprints_all, Imprints_all$imprint.11 == "PEG" & Imprints_all$imprint.14 == "PEG" & Imprints_all$imprint.17 == "PEG"& Imprints_all$imprint.21 == "PEG" )

gene_patterns(all_MEGs$feature, "Consistent MEGs") #Lol so boring
gene_patterns(all_PEGs$feature, "Consistent PEGs")
```

```{r}
#Get rid of imprinted genes that are never expressed above 5 rpm in either direction of the cross
imp.11.filter_higherthan5rpm <- subset(plot.out2.11DAP, plot.out2.11DAP$A_mean >= 5 | plot.out2.11DAP$B_mean >= 5)
imp.14.filter_higherthan5rpm <- subset(plot.out2.14DAP, plot.out2.14DAP$A_mean >= 5 | plot.out2.14DAP$B_mean >= 5)
imp.17.filter_higherthan5rpm <- subset(plot.out2.17DAP, plot.out2.17DAP$A_mean >= 5 | plot.out2.17DAP$B_mean >= 5)
imp.21.filter_higherthan5rpm <- subset(plot.out2.21DAP, plot.out2.21DAP$A_mean >= 5 | plot.out2.21DAP$B_mean >= 5)

imp.11.filter_higherthan5rpm2 <- subset(imp.11.filter_higherthan5rpm, imp.11.filter_higherthan5rpm$feature %in% Imprints_all$feature)
imp.14.filter_higherthan5rpm2 <- subset(imp.14.filter_higherthan5rpm, imp.14.filter_higherthan5rpm$feature %in% Imprints_all$feature)
imp.17.filter_higherthan5rpm2 <- subset(imp.17.filter_higherthan5rpm, imp.17.filter_higherthan5rpm$feature %in% Imprints_all$feature)
imp.21.filter_higherthan5rpm2 <- subset(imp.21.filter_higherthan5rpm, imp.21.filter_higherthan5rpm$feature %in% Imprints_all$feature)

high_imps_11and14 <- full_join(imp.11.filter_higherthan5rpm2[,c(1,13)], imp.14.filter_higherthan5rpm2[,c(1,13)], by="feature")
colnames(high_imps_11and14) <- c("feature", "RER at 11DAP", "RER at 14DAP")

high_imps_11and14and17 <- full_join(high_imps_11and14, imp.17.filter_higherthan5rpm2[,c(1,13)], by="feature")
colnames(high_imps_11and14and17) <- c("feature", "RER at 11DAP", "RER at 14DAP", "RER at 17DAP")

high_imps_all <- full_join(high_imps_11and14and17, imp.21.filter_higherthan5rpm2[,c(1,12)], by="feature")
colnames(high_imps_all) <- c("feature", "RER at 11DAP", "RER at 14DAP", "RER at 17DAP", "RER at 21DAP")

high_imps_all2 <- na.omit(high_imps_all)

#melt df for plotting RER across time for those without missing values
melted_highimps <- melt(high_imps_all2)
melted_highimps2 <- left_join(melted_highimps, Imprints_all[,c(1,6)], by ="feature")

ggplot(melted_highimps2, aes(x = variable, y = value, group = feature, fill=imprint, color=imprint)) + geom_point()+ggtitle("Imprints present in all contrasts with means above 5 rpm in either direction")+ geom_line()+ylab("Maternal Preference" )+xlab("Days After Pollination")+ ylim(0,1) +  theme_classic() + scale_color_manual(values = c("MEG"= "maroon","PEG"= "navy"))
  
```

```{r}
#Make RER heatmap 
##Make dataframe of all expressed genes' RER then remove the ones not present in the imprint all df
library(pheatmap)
library(dendsort)

plot.out_all <- full_join(plot.out.11DAP[,c(1,13)], plot.out.14DAP[,c(1,13)],by = "feature")
plot.out_all2 <- full_join(plot.out_all, plot.out.17DAP[,c(1,13)], by = "feature")
plot.out_all3 <- full_join(plot.out_all2, plot.out.21DAP[,c(1,12)], by = "feature")

plot_all_imp <- subset(plot.out_all3, plot.out_all3$feature %in% Imprints_all$feature)
colnames(plot_all_imp) <- c("feature", "11DAP", "14DAP", "17DAP", "21DAP")

callback <- function(hcl, mat){
    # Recalculate manhattan distances for reorder method
    dists <- dist(mat, method = "manhattan")

    # Perform reordering according to OLO method
    hclust_olo <- reorder(hcl, dists)
    return(hclust_olo)
}

#Create matrix for plotting
RER_matrix <- as.matrix(na.omit(plot_all_imp[,c(2:5)]))

RER_heat <- pheatmap::pheatmap(RER_matrix, show_rownames = F, cluster_cols = F, border_color = NA, main = "RER across time for imprinted genes", color = hcl.colors(50, "Batlow"), clustering_callback = callback)



plot_all_imp2 <- left_join(plot_all_imp, plot.out.11DAP[,c("feature","A_mean","B_mean")], by = "feature")
plot_all_imp3 <- left_join(plot_all_imp2, plot.out.14DAP[,c("feature","A_mean","B_mean")],by = "feature")
plot_all_imp4 <- left_join(plot_all_imp3, plot.out.17DAP[,c("feature","A_mean","B_mean")],by = "feature")
plot_all_imp5 <- left_join(plot_all_imp4, plot.out.21DAP[,c("feature","A_mean","B_mean")],by = "feature")

colnames(plot_all_imp5) <- c("feature", "RER.11","RER.14","RER.17","RER.21", "A.mean11", "B.mean11", "A.mean14", "B.mean14", "A.mean17", "B.mean17", "A.mean21", "B.mean21")

plot_all_imp6 <- plot_all_imp5

#Normalize the data before splitting
plot_all_imp6[,c(6:14)] <- plot_all_imp6[,c(6:13)]/rowMaxs(as.matrix(plot_all_imp6[,c(6:13)]))

plot_all_imp6$maternal_allele <- ifelse(plot_all_imp6$A.mean11 > plot_all_imp6$B.mean11 & plot_all_imp6$RER.11 > .7 , "A", ifelse(plot_all_imp6$B.mean11 > plot_all_imp6$A.mean11 & plot_all_imp6$RER.11 > .7, "B", ifelse(plot_all_imp6$A.mean11 > plot_all_imp6$B.mean11 & plot_all_imp6$RER.11 < .7 , "B", ifelse(plot_all_imp6$B.mean11 > plot_all_imp6$A.mean11 & plot_all_imp6$RER.11 < .7, "A", "wh"))))

table(plot_all_imp6$maternal_allele)

#Separate into two heatmaps 
forheatAmat <- subset(plot_all_imp6[,c(1,6,8,10,12)], plot_all_imp6$maternal_allele == "A")
colnames(forheatAmat) <- c("feature", "11DAP", "14DAP","17DAP","21DAP")
forheatBmat <- subset(plot_all_imp6[,c(1,7,9,11,13)], plot_all_imp6$maternal_allele == "B")
colnames(forheatBmat) <- c("feature", "11DAP", "14DAP","17DAP","21DAP")
maternal_exp <- rbind(forheatAmat,forheatBmat)

forheatApat <- subset(plot_all_imp6[,c(1,6,8,10,12)], plot_all_imp6$maternal_allele == "B")
colnames(forheatApat) <- c("feature", "11DAP", "14DAP","17DAP","21DAP")
forheatBpat <- subset(plot_all_imp6[,c(1,7,9,11,13)], plot_all_imp6$maternal_allele == "A")
colnames(forheatBpat) <- c("feature", "11DAP", "14DAP","17DAP","21DAP")
paternal_exp <- rbind(forheatApat,forheatBpat)

##attempt normalization of the reads so they can all be on one row
maternal_exp2 <- maternal_exp
rownames(maternal_exp2) <- maternal_exp2$feature
maternal_exp2$feature <- NULL
maternal_heat_matrix <- na.omit(as.matrix(maternal_exp2))

mat_heat <- pheatmap::pheatmap(maternal_heat_matrix, show_rownames = F, cluster_cols = F, border_color = NA, main = "Maternal expression across time for imprinted genes", clustering_callback = callback)


paternal_exp2 <- paternal_exp
rownames(paternal_exp2) <- paternal_exp2$feature
paternal_exp2$feature <- NULL
paternal_heat_matrix <- na.omit(as.matrix(paternal_exp2))

pat_heat <- pheatmap::pheatmap(paternal_heat_matrix, show_rownames = F, cluster_cols = F, border_color = NA, main = "Paternal expression across time for imprinted genes", clustering_callback = callback)

RER_heat
mat_heat
pat_heat


#Those still dont look right, I'm going to put the dfs back together and see how they look plotted

heat_tmp <- left_join(maternal_exp, paternal_exp, by = "feature")
heat_matrix <- na.omit(as.matrix(heat_tmp[,c(2:9)]))
pheatmap(heat_matrix, cluster_rows = T, cluster_cols = F)

#Trying one more way to make things clear
plot_all_imp5$maternal_allele <- ifelse(plot_all_imp5$A.mean11 > plot_all_imp5$B.mean11 & plot_all_imp5$RER.11 > .7 , "A", ifelse(plot_all_imp5$B.mean11 > plot_all_imp5$A.mean11 & plot_all_imp5$RER.11 > .7, "B", ifelse(plot_all_imp5$A.mean11 > plot_all_imp5$B.mean11 & plot_all_imp5$RER.11 < .7 , "B", ifelse(plot_all_imp5$B.mean11 > plot_all_imp5$A.mean11 & plot_all_imp5$RER.11 < .7, "A", "wh"))))

table(plot_all_imp5$maternal_allele)

#Separate into two heatmaps 
forheatAmat2 <- subset(plot_all_imp5[,c(1,6,8,10,12)], plot_all_imp5$maternal_allele == "A")
colnames(forheatAmat2) <- c("feature", "maternal_allele_11DAP", "maternal_allele_14DAP","maternal_allele_17DAP","maternal_allele_21DAP")
forheatBmat2 <- subset(plot_all_imp5[,c(1,7,9,11,13)], plot_all_imp5$maternal_allele == "B")
colnames(forheatBmat2) <- c("feature", "maternal_allele_11DAP", "maternal_allele_14DAP","maternal_allele_17DAP","maternal_allele_21DAP")
maternal_exp3 <- rbind(forheatAmat2,forheatBmat2)

forheatApat2 <- subset(plot_all_imp5[,c(1,6,8,10,12)], plot_all_imp5$maternal_allele == "B")
colnames(forheatApat2) <- c("feature", "paternal_allele_11DAP", "paternal_allele_14DAP","paternal_allele_17DAP","paternal_allele_21DAP")
forheatBpat2 <- subset(plot_all_imp5[,c(1,7,9,11,13)], plot_all_imp5$maternal_allele == "A")
colnames(forheatBpat2) <- c("feature", "paternal_allele_11DAP", "paternal_allele_14DAP","paternal_allele_17DAP","paternal_allele_21DAP")
paternal_exp3 <- rbind(forheatApat2,forheatBpat2)

##attempt normalization of the reads so they can all be on one row
test_this <- left_join(maternal_exp3, paternal_exp3, by = "feature")
rownames(test_this) <- test_this$feature
test_this$feature <- NULL
scale_test <- log10(test_this+1)

pheatmap::pheatmap(test_this, cluster_cols = F, show_rownames = F, scale = "row")
pheatmap::pheatmap(scale_test, cluster_cols = F, show_rownames = F)

```

```{r}
patterns_imprinting <- test_this
patterns_imprinting[is.na(patterns_imprinting)] <- 0  
#Classify Imprinting patterns by whether the other allele never turns on or it does turn on 
patterns_imprinting$feature <- rownames(patterns_imprinting)

#Add in imprinting calls from Imprints all df
  #Create imprint column in Imprints all df
Imprints_all$imprint <- ifelse(Imprints_all$imprint.11 == "MEG"|Imprints_all$imprint.14 == "MEG"|Imprints_all$imprint.17 == "MEG"|Imprints_all$imprint.21 == "MEG", "MEG", ifelse(Imprints_all$imprint.11 == "PEG"|Imprints_all$imprint.14 == "PEG"|Imprints_all$imprint.17 == "PEG"|Imprints_all$imprint.21 == "PEG", "PEG", "this shouldn't be necessary"))
patterns_imprinting2 <- left_join(patterns_imprinting, Imprints_all[,c(1,6)], by = "feature")


patterns_imprinting2$pattern <- ifelse(rowMeans(patterns_imprinting2[,c(1:4)]) < 1 & patterns_imprinting2$imprint == "PEG", "maternal allele never on", ifelse(rowMeans(patterns_imprinting2[,c(5:8)]) < 1 & patterns_imprinting2$imprint == "MEG", "paternal allele never on", ifelse(patterns_imprinting2$imprint == "MEG" & patterns_imprinting2[,c(8)] > patterns_imprinting2[,c(5)], "paternal allele goes up", ifelse(patterns_imprinting2$imprint == "MEG" & patterns_imprinting2[,c(8)] < patterns_imprinting2[,c(5)], "paternal allele goes down",ifelse(patterns_imprinting2$imprint == "PEG" & patterns_imprinting2[,c(4)] > patterns_imprinting2[,c(1)], "maternal allele goes up", ifelse(patterns_imprinting2$imprint == "PEG" & patterns_imprinting2[,c(4)] < patterns_imprinting2[,c(1)], "maternal allele goes down", "unclear"))))))



pattern_count <- patterns_imprinting2 %>% group_by(imprint, pattern) %>% dplyr::summarize(count = n())
pattern_freq <-  patterns_imprinting2 %>% group_by(imprint, pattern) %>% summarise(n=n()) %>% mutate(Freq=n/sum(n))

ggplot(pattern_count, aes(x = pattern, y=count, fill=imprint)) + geom_bar(stat="identity",position = position_dodge()) + theme_bw() +theme(axis.text.x = element_text(angle = 90)) + scale_fill_manual(values = c( "maroon", "navy"))

ggplot(pattern_freq, aes(x = pattern, y=Freq, fill=imprint)) + geom_bar(stat="identity",position = position_dodge()) + theme_bw() +theme(axis.text.x = element_text(angle = 90)) + scale_fill_manual(values = c( "maroon", "navy"))

#Lets see if any pattern is associated with endosperm preference
W22_imprints <- subset(patterns_imprinting2, grepl("Zm00004b", patterns_imprinting2$feature))
W22_pattern_count <- W22_imprints %>% group_by(imprint, pattern) %>% dplyr::summarize(count = n())
ggplot(W22_pattern_count, aes(x = pattern, y=count, fill=imprint)) + geom_bar(stat="identity",position = position_dodge()) + theme_bw() +theme(axis.text.x = element_text(angle = 90)) + scale_fill_manual(values = c( "maroon", "navy"))

W22_atlas <- read.table("//las-dfs-01.las.iastate.edu/lss/research/sna-lab/kmh/projects/mdr1/W22_endo_pref_call.txt")
W22_atlas$feature <- rownames(W22_atlas)

W22_imprints_pref <- left_join(W22_imprints, W22_atlas, by= "feature")

W22_pattern_count <- W22_imprints_pref %>% group_by(preference,imprint, pattern) %>% dplyr::summarize(count = n())
ggplot(W22_pattern_count, aes(x = pattern, y=count, fill=imprint)) + geom_bar(stat="identity",position = position_dodge()) + theme_bw() +theme(axis.text.x = element_text(angle = 90)) + scale_fill_manual(values = c( "maroon", "navy")) + facet_wrap(.~preference)
#Not much of a difference,  I don't believe there is a connection between pattern and tissue

#Examples of different patterns
    #First paternal allele goes up 
my_gene("Zm00004b020603")
    #Next, paternal allele never on
my_gene("Zm00001eb359000")
    # paternal allele goes down
my_gene("Zm00001eb265890")
    #maternal goes up
my_gene("Zm00004b030050")
    #maternal allele never on 
my_gene("Zm00001eb407830")
    #maternal allele goes down (had highest values for PEGs)
my_gene("Zm00001eb323510")

#Looking for an outlier
my_gene("Zm00004b027397") #This turns out to be a pyrophosphatase 


#Which pattern do consistent imprints belong to generally?
    #Add column to Imprinting_all df specifying consistency
Imprints_all$consistency <- ifelse(Imprints_all$imprint.11 == Imprints_all$imprint.14 & Imprints_all$imprint.14 == Imprints_all$imprint.17 & Imprints_all$imprint.17 == Imprints_all$imprint.21,"consistent", "transient")

patterns_imp_plusconsistency <- left_join(patterns_imprinting2, Imprints_all[,c(1,7)], by ="feature")

pattern_count <- patterns_imp_plusconsistency %>% group_by(consistency,imprint, pattern) %>% dplyr::summarize(count = n())
pattern_freq <-  patterns_imp_plusconsistency %>% group_by(imprint, consistency,pattern) %>% summarise(n=n()) %>% mutate(Freq=n/sum(n))

ggplot(pattern_count, aes(x = pattern, y=count, fill=imprint)) + geom_bar(stat="identity",position = position_dodge()) + theme_bw() +theme(axis.text.x = element_text(angle = 90)) + scale_fill_manual(values = c( "maroon", "navy")) + facet_wrap(.~consistency)

ggplot(pattern_freq, aes(x = pattern, y=Freq, fill=consistency)) + geom_bar(stat="identity",position = position_dodge()) + theme_bw() +theme(axis.text.x = element_text(angle = 90)) + scale_fill_manual(values = c( "sienna2", "slateblue4"))

#variance
pattern.variance <- aov(pattern ~ consistency, data = pattern_count)


#Before running this next line syntplot.Rmd needs to have run to create the function and files needed. 

#Examples of different patterns across lines
    #First paternal allele goes up 
syntplot_Function("Zm00004b020603")
    #Next, paternal allele never on
syntplot_Function("Zm00001eb359000")
    # paternal allele goes down
syntplot_Function("Zm00001eb265890")
    #maternal goes up
syntplot_Function("Zm00004b030050")
    #maternal allele never on 
syntplot_Function("Zm00001eb407830")
    #maternal allele goes down (had highest values for PEGs)
syntplot_Function("Zm00001eb323510")

#Looking for an outlier
syntplot_Function("Zm00004b027397") #This turns out to be a pyrophosphatase 

maternal_down <- subset(patterns_imprinting2$feature, patterns_imprinting2$pattern == "maternal allele goes down")

peg1 <- NAM_tally(maternal_down, "maternal allele goes down")

maternal_off <- subset(patterns_imprinting2$feature, patterns_imprinting2$pattern == "maternal allele never on")

peg2 <- NAM_tally(maternal_off,"maternal allele never on")

maternal_up <- subset(patterns_imprinting2$feature, patterns_imprinting2$pattern == "maternal allele goes up")

peg3 <- NAM_tally(maternal_up,"maternal allele goes up")

paternal_down <- subset(patterns_imprinting2$feature, patterns_imprinting2$pattern == "paternal allele goes down")

meg1 <- NAM_tally(paternal_down, "paternal allele goes down")

paternal_off <- subset(patterns_imprinting2$feature, patterns_imprinting2$pattern == "paternal allele never on")

meg2 <- NAM_tally(paternal_off, "paternal allele never on")

paternal_up <- subset(patterns_imprinting2$feature, patterns_imprinting2$pattern == "paternal allele goes up")

meg3 <- NAM_tally(paternal_up, "paternal allele goes up")

grid.arrange(peg1,peg2,peg3,meg1,meg2,meg3,nrow=3)

```

```{r}
#separate MEGs and PEGs by genome they belong to

B73_imprints <- subset(patterns_imprinting2, grepl("Zm00001eb", patterns_imprinting2$feature))
W22_imprints <- subset(patterns_imprinting2, grepl("Zm00004b", patterns_imprinting2$feature))

table(B73_imprints$imprint, B73_imprints$pattern)
table(W22_imprints$imprint, W22_imprints$pattern)

W22maternal_down <- subset(W22_imprints$feature, W22_imprints$pattern == "maternal allele goes down")
W22maternal_off <- subset(W22_imprints$feature, W22_imprints$pattern == "maternal allele never on")
W22maternal_up <- subset(W22_imprints$feature, W22_imprints$pattern == "maternal allele goes up")
W22paternal_down <- subset(W22_imprints$feature, W22_imprints$pattern == "paternal allele goes down")
W22paternal_off <- subset(W22_imprints$feature, W22_imprints$pattern == "paternal allele never on")
W22paternal_up <- subset(W22_imprints$feature, W22_imprints$pattern == "paternal allele goes up")

B73maternal_down <- subset(B73_imprints$feature, B73_imprints$pattern == "maternal allele goes down")
B73maternal_off <- subset(B73_imprints$feature, B73_imprints$pattern == "maternal allele never on")
B73maternal_up <- subset(B73_imprints$feature, B73_imprints$pattern == "maternal allele goes up")
B73paternal_down <- subset(B73_imprints$feature, B73_imprints$pattern == "paternal allele goes down")
B73paternal_off <- subset(B73_imprints$feature, B73_imprints$pattern == "paternal allele never on")
B73paternal_up <- subset(B73_imprints$feature, B73_imprints$pattern == "paternal allele goes up")

#Go term enrichment analysis
#load packages
library(topGO)
library(tidyverse)
library(scales)

fname = '//las-dfs-01.las.iastate.edu/lss/research/sna-lab/kmh/projects/timepoint/B73_GO.out'
go <-
  read.table(
    fname,
    quote = '"',
    sep = "\t",
    header = TRUE,
    colClasses = c('goid' = 'character', 'qpid' = 'character')
  )
# some filtering
go_filt <- go[(go$ARGOT_PPV > 0.5), ]

go_filt$goid <- paste0('GO:', go_filt$goid)

panzer_to_golist <- function(panzer_df){
  go_df <- aggregate( goid ~ qpid, data=panzer_df, FUN=c)
  structure(go_df$goid, .Names=go_df$qpid)
}
all_golist <- panzer_to_golist(go_filt)

gene_list <- as.data.frame(rownames(counts_kmeans))

gene_list <- colnames("genenames")

#Create individual gene lists for the clusters
up14 <- as.data.frame(rownames(up_14))
colnames(up14) <- "genenames"
up14 <- up14 %>% mutate(myInput = "yes")
write.table(up14[,c(1)], "//las-dfs-01.las.iastate.edu/lss/research/sna-lab/kmh/projects/timepoint/up14.txt")

up17 <- as.data.frame(rownames(up_17))
colnames(up17) <- "genenames"
up17 <- up17 %>% mutate(myInput = "yes")
write.table(up17[,c(1)], "//las-dfs-01.las.iastate.edu/lss/research/sna-lab/kmh/projects/timepoint/up17.txt")

up21 <- as.data.frame(rownames(up_21))
colnames(up21) <- "genenames"
up21 <- up21 %>% mutate(myInput = "yes")
write.table(up21[,c(1)], "//las-dfs-01.las.iastate.edu/lss/research/sna-lab/kmh/projects/timepoint/up21.txt")


down14 <- as.data.frame(rownames(down_14))
colnames(down14) <- "genenames"
down14 <- down14 %>% mutate(myInput = "yes")
write.table(down14[,c(1)], "//las-dfs-01.las.iastate.edu/lss/research/sna-lab/kmh/projects/timepoint/down14.txt")

down17 <- as.data.frame(rownames(down_17))
colnames(down17) <- "genenames"
down17 <- down17 %>% mutate(myInput = "yes")
write.table(down17[,c(1)], "//las-dfs-01.las.iastate.edu/lss/research/sna-lab/kmh/projects/timepoint/down17.txt")

down21 <- as.data.frame(rownames(down_21))
colnames(down21) <- "genenames"
down21 <- down21 %>% mutate(myInput = "yes")
write.table(down21[,c(1)], "//las-dfs-01.las.iastate.edu/lss/research/sna-lab/kmh/projects/timepoint/down21.txt")


#Get all genes from annotation list
gene_names <- names(all_golist)
gene_names2 <- substr(gene_names,1,15)
myGenes <- as.data.frame(gene_names2)
colnames(myGenes) <- "genenames"

#Get dfs ready
myGenes_1 <- left_join(myGenes,up14)
  myGenes_1$myInput <- myGenes_1$myInput %>%
  replace_na('no')
  up14_GO <- factor(as.integer(myGenes_1$myInput == 'yes'))
  names(up14_GO) <- gene_names

myGenes_2 <- left_join(myGenes,up17)
  myGenes_2$myInput <- myGenes_2$myInput %>%
  replace_na('no')
  up17_GO <- factor(as.integer(myGenes_2$myInput == 'yes'))
  names(up17_GO) <- gene_names
  
myGenes_3 <- left_join(myGenes,up21)
  myGenes_3$myInput <- myGenes_3$myInput %>%
  replace_na('no')
  up21_GO <- factor(as.integer(myGenes_3$myInput == 'yes'))
  names(up21_GO) <- gene_names

myGenes_4 <- left_join(myGenes,down14)
  myGenes_4$myInput <- myGenes_4$myInput %>%
  replace_na('no')
  down14_GO <- factor(as.integer(myGenes_4$myInput == 'yes'))
  names(down14_GO) <- gene_names
  
myGenes_5 <- left_join(myGenes,down17)
  myGenes_5$myInput <- myGenes_5$myInput %>%
  replace_na('no')
  down17_GO <- factor(as.integer(myGenes_5$myInput == 'yes'))
  names(down17_GO) <- gene_names

myGenes_6 <- left_join(myGenes,down21)
  myGenes_6$myInput <- myGenes_6$myInput %>%
  replace_na('no')
  down21_GO <- factor(as.integer(myGenes_6$myInput == 'yes'))
  names(down21_GO) <- gene_names


#Set up GO enrichment analysis 
run_topGO <- function(gene_list, ontology, gene2GO_list){
  topGO_out <- new("topGOdata", ontology = ontology, allGenes = gene_list,
                    annot = annFUN.gene2GO, gene2GO = gene2GO_list)
  topGO_out
  fishers_result <- runTest(topGO_out, algorithm = "elim", statistic = "fisher")
  fishers_table <- GenTable(topGO_out, Fishers = fishers_result, useLevels = TRUE)
  fishers_table$Ontology <- ontology
  fishers_table$Fishers <- as.numeric(fishers_table$Fishers)
  fishers_table
}


#Run TopGO on all ontology terms and clusters 
up14_BP <- run_topGO(up14_GO, "BP", all_golist)
up14_MF <- run_topGO(up14_GO, "MF", all_golist)
up14_CC <- run_topGO(up14_GO, "CC", all_golist)
up14GO <- (rbind(up14_BP, up14_MF, up14_CC))
up14GO <- up14GO[order(up14GO$Fishers),]


up17_BP <- run_topGO(up17_GO, "BP", all_golist)
up17_MF <- run_topGO(up17_GO, "MF", all_golist)
up17_CC <- run_topGO(up17_GO, "CC", all_golist)
up17GO <- (rbind(up17_BP, up17_MF, up17_CC))
up17GO <- up17GO[order(up17GO$Fishers),]

up21_BP <- run_topGO(up21_GO, "BP", all_golist)
up21_MF <- run_topGO(up21_GO, "MF", all_golist)
up21_CC <- run_topGO(up21_GO, "CC", all_golist)
up21GO <- (rbind(up21_BP, up21_MF, up21_CC))
up21GO <- up21GO[order(up21GO$Fishers),]

down14_BP <- run_topGO(down14_GO, "BP", all_golist)
down14_MF <- run_topGO(down14_GO, "MF", all_golist)
down14_CC <- run_topGO(down14_GO, "CC", all_golist)
down14GO <- (rbind(down14_BP, down14_MF, down14_CC))
down14GO <- down14GO[order(down14GO$Fishers),]

down17_BP <- run_topGO(down17_GO, "BP", all_golist)
down17_MF <- run_topGO(down17_GO, "MF", all_golist)
down17_CC <- run_topGO(down17_GO, "CC", all_golist)
down17GO <- (rbind(down17_BP, down17_MF, down17_CC))
down17GO <- down17GO[order(down17GO$Fishers),]

down21_BP <- run_topGO(down21_GO, "BP", all_golist)
down21_MF <- run_topGO(down21_GO, "MF", all_golist)
down21_CC <- run_topGO(down21_GO, "CC", all_golist)
down21GO <- (rbind(down21_BP, down21_MF, down21_CC))
down21GO <- down21GO[order(down21GO$Fishers),]


#Make function to create bar plots of enriched terms
myBarPlot <- function(table , term ) {
  table <- table %>% mutate(word = case_when(Significant == 1 ~ "hit",
                                             TRUE ~ "hits"))
  ggplot(table, aes(reorder(Term, -log10(Fishers)),
                    -log10(Fishers), fill = Ontology)) +
    geom_bar(stat = 'identity') +
    scale_fill_manual(values = c(
      "BP" = "#9531b8",
      "MF" = "#9dd745",
      "CC" = "#da4218"
    )) +
    geom_text(aes(label = paste(Significant, word, "of", Annotated)),
              hjust = 1.2,
              color = "white") +
    theme_classic(base_size = 14) +
    ylab("- log10 p-value") +
    xlab("") +
    labs(title = paste0('GO Analysis (', term, ")")) +
    scale_y_continuous(expand = expansion(mult = c(0, .1)),
                       breaks = scales::pretty_breaks()) +
    theme(legend.position = "none") +
    coord_flip()
}

up14_plot <- myBarPlot(table = up14GO, term = "Upregulated at 14DAP")
up17_plot <- myBarPlot(table = up17GO, term = "Upregulated at 17DAP")
up21_plot <- myBarPlot(table = up21GO, term = "Upregulated at 21DAP")
down14_plot <- myBarPlot(table = down14GO, term = "Downregulated at 14DAP")
down17_plot <- myBarPlot(table = down17GO, term = "Downregulated at 17DAP")
down21_plot <- myBarPlot(table = down21GO, term = "Downregulated at 21DAP")


up14_plot
up17_plot
up21_plot
down14_plot
down17_plot
down21_plot

#Check for common terms
tmp <- bind_rows(up14GO,up17GO,up21GO,down14GO,down17GO,down21GO, .id = "Direction")
tmp$Direction <- ifelse(tmp$Direction == 1|tmp$Direction == 2|tmp$Direction == 3,"up","down")
tmp2 <- tmp %>% group_by(Direction,Term) %>% summarize(count = n())
```

