---
title: "R Notebook"
output: html_notebook
---

This is an [R Markdown](http://rmarkdown.rstudio.com) Notebook. When you execute code within the notebook, the results appear beneath the code. 

Try executing this chunk by clicking the *Run* button within the chunk or by placing your cursor inside it and pressing *Ctrl+Shift+Enter*. 

```{r}
library(ggplot2)
library(DESeq2)
library(MaizePal)
library(rstatix)
library(pheatmap)
library(viridisLite)
library(viridis)
library(ggforce)
library(ggExtra)
library(gridExtra)
#library(MatrixGenerics)
library(matrixStats)
library("ggalluvial")
#library(DelayedMatrixStats)
library(dplyr)
```

```{r}
counts <- read.table("//las-dfs-01.las.iastate.edu/lss/research/sna-lab/kmh/projects/timepoint/combined_timepoint_counts.txt",header=T,stringsAsFactors = F)

#Change row name to the gene ID
rownames(counts) <- counts$ID
counts$ID <- NULL

#Normalize counts into reads per million
counts.rpm <- counts
for(i in 1:26){
  counts.rpm[,i] <- (counts[,i]/sum(counts[,i])) * 1e6
}
##Create heatmap to see if groups cluster together
counts.rpm2 <- subset(counts.rpm, grepl("Zm", rownames(counts.rpm)))
counts.rpm.3 <- counts.rpm2[,1:25]

data_subset <- as.matrix(counts.rpm.3[rowSums(counts.rpm.3)>1000,])

pheatmap((data_subset/rowMaxs(data_subset)),border_color = NA,fontsize = 8)

##Correlation heatmap
library(NMF)
corr <- cor(counts.rpm.3)
aheatmap(corr,main = "Correlation beween samples",labCol=substr(colnames(counts.rpm.3),1,4),labRow=substr(colnames(counts.rpm.3),1,4))


#Subset only the counts
counts2 <- subset(counts, grepl("Zm", rownames(counts)))
counts.de <- counts2[,1:25]
rownames(counts.de) <- row.names(counts2)

#Get rid of rows with <10 reads in non-normalized set
counts.de.keep <- subset(counts.de,rowSums(counts.de) >= 10) 

#Make sure the mutant and wildtype have at least 1 read in each of the 3 reps in normalized set
#counts.de.keep <- subset(counts.de,rowMeans(counts.rpm[,1:3]) >= 0.5 | rowMeans(counts.rpm[,4:6]) >= 0.5) 

#Set sample information for DE calculation
sample.info <- as.data.frame(cbind(paste(names(counts.de.keep)),c("BW11", "BW11", "BW11", "WB11", "WB11", "WB11", "BW14", "BW14", "BW14", "WB14", "WB14", "WB14", "BW17", "BW17", "BW17", "WB17", "WB17", "WB17", "BW21", "BW21", "BW21", "WB21", "WB21", "BW25", "WB25")))
rownames(sample.info) <- sample.info[,1]
sample.info$timepoint <- substr(sample.info$V2,3,4)
sample.info$genotype <- substr(sample.info$V1,1,2)

#Run DESeq
dds <- DESeqDataSetFromMatrix(countData = counts.de.keep,colData = sample.info, design = ~ timepoint + genotype + timepoint:genotype)
dds2 <- DESeq(dds, test = "LRT", reduced = ~ timepoint + genotype)
keep <- rowSums(counts(dds2)) >1
dds2 <- dds2[keep,]
#Get names of different tests performed
resultsNames(dds2)
#make results tables
res_17v11 <- results(dds2, name = "timepoint_17_vs_11", alpha = .001 )
res_14v11 <- results(dds2, name = "timepoint_14_vs_11", alpha = .001 )
res_21v11 <- results(dds2, name = "timepoint_21_vs_11", alpha = .001 )
res_WBvBW <- results(dds2, name = "genotype_WB_vs_BW", alpha = .001 )
res_int <- results(dds2, name = "Intercept", alpha = .001)
#Coerce results to dataframes
out_17v11<- data.frame(res_17v11)
out_14v11<- data.frame(res_14v11)
out_21v11<- data.frame(res_21v11)
out_WBvBW<- data.frame(res_WBvBW)
out_int <- data.frame(res_int)

rld <- rlog(dds2, blind=FALSE)
a <- plotPCA(rld,intgroup="V2",ntop=10000)
a

b <- plotPCA(rld,intgroup="genotype")
b

#write.table(counts.rpm.3, file = "//las-dfs-01.las.iastate.edu/lss/research/sna-lab/kmh/projects/timepoint/timepoint.rpm.txt")

```

```{r}

#Get rid of low counts

counts.rpm.keep <- subset(counts.rpm.3,rowSums(counts.rpm.3) >= 10) 

#We want to know whether things are genes or TEs so we define that with this command, which asks if the rownames are longer than 20 characters (genes are 14 or 15 characters long) or if it contains the label 'TE' 

counts.rpm.keep$feature_type <- ifelse(nchar(rownames(counts.rpm.keep)) > 20 |grepl("TE",rownames(counts.rpm.keep)), "TE","gene")# Label features as TEs or genes
table(counts.rpm.keep$feature_type)

counts.rpm.keep$genome <-ifelse(grepl("Zm00001e|B73",rownames(counts.rpm.keep)),"B73",ifelse(grepl("Zm00004b|W22",rownames(counts.rpm.keep)), "W22","NA")) # Label which genome the feature annotation is from
table(counts.rpm.keep$genome)


#Subset one contrast
imp.11DAP.rpm <- counts.rpm.keep[,c(1:6,26,27)]
#Give the contrast a name, in this case it would be the timepoint
imp.11DAP.rpm$contrast <- "11DAP"
#Create a new column to simplify using which genome the gene originated from 
imp.11DAP.rpm$type <- ifelse(imp.11DAP.rpm$genome == "B73","A","B")
#transfer rownames
imp.11DAP.rpm$feature <- rownames(imp.11DAP.rpm)

#Rename columns, reduce the 'cross' columns for comparisons with fewer columns
names(imp.11DAP.rpm) <- c("cross1","cross2","cross3", "cross4","cross5","cross6","feature_type","genome","contrast","type","feature")

#Create the A_mean and B_mean, in this example B73 is the first 3 columns of the subset dataframe, so I ask for the mean of those three columns as the A_mean, and the other 3 as the B_mean
imp.11DAP.rpm$A_mean <- rowMeans(imp.11DAP.rpm[,1:3])
imp.11DAP.rpm$B_mean <- rowMeans(imp.11DAP.rpm[,4:6])

#here is the calculation of maternal preference
imp.11DAP.rpm$maternal_preference <- ifelse(imp.11DAP.rpm$type == "A",imp.11DAP.rpm$A_mean/(imp.11DAP.rpm$B_mean + imp.11DAP.rpm$A_mean) ,imp.11DAP.rpm$B_mean/(imp.11DAP.rpm$B_mean + imp.11DAP.rpm$A_mean))

#Subset one contrast
imp.14DAP.rpm <- counts.rpm.keep[,c(7:12,26,27)]
#Give the contrast a name, in this case it would be the timepoint
imp.14DAP.rpm$contrast <- "14DAP"
#Create a new column to simplify using which genome the gene originated from 
imp.14DAP.rpm$type <- ifelse(imp.14DAP.rpm$genome == "B73","A","B")
#transfer rownames
imp.14DAP.rpm$feature <- rownames(imp.14DAP.rpm)

#Rename columns, reduce the 'cross' columns for comparisons with fewer columns
names(imp.14DAP.rpm) <- c("cross1","cross2","cross3", "cross4","cross5","cross6","feature_type","genome","contrast","type","feature")

#Create the A_mean and B_mean, in this example B73 is the first 3 columns of the subset dataframe, so I ask for the mean of those three columns as the A_mean, and the other 3 as the B_mean
imp.14DAP.rpm$A_mean <- rowMeans(imp.14DAP.rpm[,1:3])
imp.14DAP.rpm$B_mean <- rowMeans(imp.14DAP.rpm[,4:6])

#here is the calculation of maternal preference
imp.14DAP.rpm$maternal_preference <- ifelse(imp.14DAP.rpm$type == "A",imp.14DAP.rpm$A_mean/(imp.14DAP.rpm$B_mean + imp.14DAP.rpm$A_mean) ,imp.14DAP.rpm$B_mean/(imp.14DAP.rpm$B_mean + imp.14DAP.rpm$A_mean))

#Subset one contrast
imp.17DAP.rpm <- counts.rpm.keep[,c(13:18,26,27)]
#Give the contrast a name, in this case it would be the timepoint
imp.17DAP.rpm$contrast <- "17DAP"
#Create a new column to simplify using which genome the gene originated from 
imp.17DAP.rpm$type <- ifelse(imp.17DAP.rpm$genome == "B73","A","B")
#transfer rownames
imp.17DAP.rpm$feature <- rownames(imp.17DAP.rpm)

#Rename columns, reduce the 'cross' columns for comparisons with fewer columns
names(imp.17DAP.rpm) <- c("cross1","cross2","cross3", "cross4","cross5","cross6","feature_type","genome","contrast","type","feature")

#Create the A_mean and B_mean, in this example B73 is the first 3 columns of the subset dataframe, so I ask for the mean of those three columns as the A_mean, and the other 3 as the B_mean
imp.17DAP.rpm$A_mean <- rowMeans(imp.17DAP.rpm[,1:3])
imp.17DAP.rpm$B_mean <- rowMeans(imp.17DAP.rpm[,4:6])

#here is the calculation of maternal preference
imp.17DAP.rpm$maternal_preference <- ifelse(imp.17DAP.rpm$type == "A",imp.17DAP.rpm$A_mean/(imp.17DAP.rpm$B_mean + imp.17DAP.rpm$A_mean) ,imp.17DAP.rpm$B_mean/(imp.17DAP.rpm$B_mean + imp.17DAP.rpm$A_mean))

#Subset one contrast
imp.21DAP.rpm <- counts.rpm.keep[,c(19:23,26,27)]
#Give the contrast a name, in this case it would be the timepoint
imp.21DAP.rpm$contrast <- "21DAP"
#Create a new column to simplify using which genome the gene originated from 
imp.21DAP.rpm$type <- ifelse(imp.21DAP.rpm$genome == "B73","A","B")
#transfer rownames
imp.21DAP.rpm$feature <- rownames(imp.21DAP.rpm)

#Rename columns, reduce the 'cross' columns for comparisons with fewer columns
names(imp.21DAP.rpm) <- c("cross1","cross2","cross3", "cross4","cross5","feature_type","genome","contrast","type","feature")

#Create the A_mean and B_mean, in this example B73 is the first 3 columns of the subset dataframe, so I ask for the mean of those three columns as the A_mean, and the other 3 as the B_mean
imp.21DAP.rpm$A_mean <- rowMeans(imp.21DAP.rpm[,1:3])
imp.21DAP.rpm$B_mean <- rowMeans(imp.21DAP.rpm[,4:5])

#here is the calculation of maternal preference
imp.21DAP.rpm$maternal_preference <- ifelse(imp.21DAP.rpm$type == "A",imp.21DAP.rpm$A_mean/(imp.21DAP.rpm$B_mean + imp.21DAP.rpm$A_mean) ,imp.21DAP.rpm$B_mean/(imp.21DAP.rpm$B_mean + imp.21DAP.rpm$A_mean))

imp.21.mat <- imp.21DAP.rpm[,c("feature","maternal_preference","contrast")]
imp.17.mat <- imp.17DAP.rpm[,c("feature","maternal_preference","contrast")]
imp.14.mat <- imp.14DAP.rpm[,c("feature","maternal_preference","contrast")]
imp.11.mat <- imp.11DAP.rpm[,c("feature","maternal_preference","contrast")]


imp.mat1 <- right_join(imp.11.mat,imp.17.mat, by = "feature")
imp.mat2 <- right_join(imp.mat1,imp.14.mat, by = "feature")
imp.mat.all <- right_join(imp.mat2, imp.21.mat, by ="feature")

#rename columns
names(imp.mat.all) <- c("feature","mat_pref11","11DAP","mat_pref17","17DAP","mat_pref14","14DAP","mat_pref21","21DAP")
#prep for melting 
prep <- imp.mat.all[,c(1,2,4,6,8)]

#Make df for including mean rpms for each timepoint
imp.21.rpm <- imp.21DAP.rpm[,c("feature","A_mean","B_mean","genome","contrast")]
imp.17.rpm <- imp.17DAP.rpm[,c("feature","A_mean","B_mean","contrast")]
imp.14.rpm <- imp.14DAP.rpm[,c("feature","A_mean","B_mean","contrast")]
imp.11.rpm <- imp.11DAP.rpm[,c("feature","A_mean","B_mean","contrast")]



imp.rpm1 <- right_join(imp.11.rpm,imp.14.rpm, by = "feature")
imp.rpm2 <- right_join(imp.rpm1,imp.17.rpm, by = "feature")
imp.rpm.all <- right_join(imp.rpm2, imp.21.rpm, by ="feature")

#rename columns
names(imp.rpm.all) <- c("feature","A_mean11","B_mean11","11DAP","A_mean14","B_mean14","14DAP","A_mean17","B_mean17","17DAP","A_mean21","B_mean21","genome","21DAP")

#prep for melt
prep2 <- imp.rpm.all[,c(1,2,3,5,6,8,9,11:13)]
#melt
imp.rpm.melted <- melt(prep2, id.vars = c("feature","genome"),varnames = c("A_mean","B_mean"))
#Create new columns for coloring
imp.rpm.melted$Timepoint <- ifelse(grepl("11", imp.rpm.melted$variable), "11 DAP",ifelse(grepl("14", imp.rpm.melted$variable), "14 DAP",ifelse(grepl("17", imp.rpm.melted$variable), "17 DAP",ifelse(grepl("21", imp.rpm.melted$variable), "21 DAP", "na"))))
imp.rpm.melted$group <- ifelse(grepl("A_mean", imp.rpm.melted$variable), "A",ifelse(grepl("B_mean", imp.rpm.melted$variable), "B","na"))
imp.rpm.melted$group2 <- ifelse(imp.rpm.melted$group=="A" & imp.rpm.melted$genome == "B73", "maternal", ifelse(imp.rpm.melted$group=="B"& imp.rpm.melted$genome=="W22", "maternal", "paternal"))




imp.all.melted <- melt(prep, id.vars = "feature",value.name = "maternal_preference")

imp.all.melted$Timepoint <- ifelse(grepl("11", imp.all.melted$variable), "11 DAP",ifelse(grepl("14", imp.all.melted$variable), "14 DAP",ifelse(grepl("17", imp.all.melted$variable), "17 DAP",ifelse(grepl("21", imp.all.melted$variable), "21 DAP", "na"))))

imp.all.melted$exp <- ifelse(imp.all.melted$maternal_preference >= .8, "possible MEG", ifelse(imp.all.melted$maternal_preference < .4, "possible PEG", "biparental" ))

library(readxl)
library(tidyverse)
library(ggplot2)
library(dplyr)
library(tidyr)
library(ggExtra)

gene_list<-data.frame(unique(imp.all.melted$feature))
my_gene<- function(gene){                         #function will take in a value(gene)
  gene_list<-data.frame(unique(imp.all.melted$feature)) 
  
  gene_data <- imp.all.melted[imp.all.melted$feature == gene, ]   #search through the original data and eliminate all the rows that do not mention the gene. left with gene data
  gene_exp <- imp.rpm.melted[imp.rpm.melted$feature == gene,]
  gene_mat_pref <- gene_data %>% ggplot(aes(x=Timepoint , y= maternal_preference, group =1, color = exp))+
    geom_point()+ggtitle(gene)+ geom_line()+ylab("Maternal Preference" )+xlab("Days After Pollination")+ scale_color_manual("Predicted expression",values = c("possible MEG" = "maroon", "possible PEG" = "navy", "biparental" = "darkgrey")) + ylim(0,1)
  gene_expression <- gene_exp %>% ggplot(aes(x=Timepoint, y=value, color=group2)) + geom_point() +ylab("Normalized Expression in RPM") + xlab("Days After Pollination")
grid.arrange(gene_mat_pref,gene_expression, ncol = 1)
}

```


Haash out this block if not needed since it was for testing the plots
#```{r}
a <- my_gene("Zm00001eb166940")
b <- my_gene("Zm00001eb313790")
c <- my_gene("Zm00001eb166950")
d <- my_gene("Zm00001eb267570")
e <- my_gene("Zm00001eb030160")
f <- my_gene("Zm00001eb303160")
g <- my_gene("Zm00001eb382030")
h <- my_gene("Zm00001eb303130")
i <- my_gene("Zm00001eb281380")
j <- my_gene("Zm00001eb166600")
k <- my_gene("Zm00001eb166660")
l <- my_gene("Zm00001eb166670")
m <- my_gene("Zm00001eb315100")

a
b
c
d
e
f
g
h
i
j
k
l
m


n <- my_gene("Zm00001eb166610")
o <- my_gene("Zm00001eb313800")
p <- my_gene("Zm00001eb030130")
q <- my_gene("Zm00001eb355270")
r <- my_gene("Zm00001eb166580")
s <- my_gene("Zm00001eb171940")

n
o
p
q
r
s


t <-my_gene("Zm00004b020298") 
u <- my_gene("Zm00004b036075")
v <- my_gene("Zm00004b036199")
w <- my_gene("Zm00004b026780")
x <- my_gene("Zm00004b020299")
y <- my_gene("Zm00004b028826")
z <- my_gene("Zm00004b036076")
aa <- my_gene("Zm00004b002718")
ab <- my_gene("Zm00004b032428")
ac <- my_gene("Zm00004b020257")
ad <-my_gene("Zm00004b002717")
ae <- my_gene("Zm00004b020270")
af <- my_gene("Zm00004b020267")
t
u
v
w
x
y
z
aa
ab
ac
ad
ae
af

my_gene("Zm00001eb010420")

my_gene("Zm00004b027627")


#Take prep file and use it to see if there are any genes that have high variation in maternal preference
row.names(prep) <- prep$feature
prep$feature <- NULL

RowVar <- function(x, ...) {
  rowSums((x - rowMeans(x, ...))^2, ...)/(dim(x)[2] - 1)
}
prep$coefficient_variation <- NULL
prep$row_mean <- NULL
prep$sd <- NULL

prep$row_mean <- rowMeans(prep)
prep$sd <- apply(prep[,c(1:4)], 1, sd)

prep$coefficient_variation <- prep$sd/prep$row_mean

my_gene("Zm00004b014223")
my_gene("RLG00003Zm00004b10883")
my_gene("Zm00001eb173090") #Fie1 in B73
my_gene("Zm00004b020902") #Fie1 in W22
my_gene("RLX29535Zm00004b00001") #MEG at 14DAP in BW and WP
my_gene("DHH00002Zm00004b00551")
my_gene("Zm00004b038376")
my_gene("Zm00004b008136")

#Checking for Vital
my_gene("Zm00001eb0166950")
my_gene("Zm00001eb040700") #Arid9
my_gene("Zm00001eb304730") #VIM104, not imprinted here but its not surprising since it isn't abundant in the previous W22 and B73 crosses
my_gene("Zm00001eb279420") #Conserved MEG
my_gene("Zm00001eb271490")#MEZ1 

#```

```{r}
#Make alluvial plot of results
library(tidyr)
library(ggalluvial)
library(stringr)

imp.rpm.all$pref_11 <- ifelse(imp.rpm.all$genome =="B73", imp.rpm.all$A_mean11/(imp.rpm.all$B_mean11 + imp.rpm.all$A_mean11), imp.rpm.all$B_mean11/(imp.rpm.all$B_mean11 + imp.rpm.all$A_mean11))
imp.rpm.all$pref_14 <- ifelse(imp.rpm.all$genome =="B73", imp.rpm.all$A_mean14/(imp.rpm.all$B_mean14 + imp.rpm.all$A_mean14), imp.rpm.all$B_mean14/(imp.rpm.all$B_mean14 + imp.rpm.all$A_mean14))
imp.rpm.all$pref_17 <- ifelse(imp.rpm.all$genome =="B73", imp.rpm.all$A_mean17/(imp.rpm.all$B_mean17 + imp.rpm.all$A_mean17), imp.rpm.all$B_mean17/(imp.rpm.all$B_mean17 + imp.rpm.all$A_mean17))
imp.rpm.all$pref_21 <- ifelse(imp.rpm.all$genome =="B73", imp.rpm.all$A_mean21/(imp.rpm.all$B_mean21 + imp.rpm.all$A_mean21), imp.rpm.all$B_mean21/(imp.rpm.all$B_mean21 + imp.rpm.all$A_mean21))

imp.rpm.all$exp11 <- ifelse(imp.rpm.all$pref_11 >= .9 , "maternal", ifelse(imp.rpm.all$pref_11 <= .3, "paternal",ifelse(imp.rpm.all$pref_11 > .3 & imp.rpm.all$pref_11 <.4, "bordering paternal", ifelse(imp.rpm.all$pref_11 >.8 & imp.rpm.all$pref_11 < .9, "bordering maternal", "biparental"))))

imp.rpm.all$exp14 <- ifelse(imp.rpm.all$pref_14 >= .9 , "maternal", ifelse(imp.rpm.all$pref_14 <= .3, "paternal",ifelse(imp.rpm.all$pref_14 > .3 & imp.rpm.all$pref_14 <.4, "bordering paternal", ifelse(imp.rpm.all$pref_14 >.8 & imp.rpm.all$pref_14 < .9, "bordering maternal", "biparental"))))

imp.rpm.all$exp17 <- ifelse(imp.rpm.all$pref_17 >= .9 , "maternal", ifelse(imp.rpm.all$pref_17 <= .3, "paternal",ifelse(imp.rpm.all$pref_17 > .3 & imp.rpm.all$pref_17 <.4, "bordering paternal", ifelse(imp.rpm.all$pref_17 >.8 & imp.rpm.all$pref_17 < .9, "bordering maternal", "biparental"))))

imp.rpm.all$exp21 <- ifelse(imp.rpm.all$pref_21 >= .9 , "maternal", ifelse(imp.rpm.all$pref_21 <= .3, "paternal",ifelse(imp.rpm.all$pref_21 > .3 & imp.rpm.all$pref_21 <.4, "bordering paternal", ifelse(imp.rpm.all$pref_21 >.8 & imp.rpm.all$pref_21 < .9, "bordering maternal", "biparental"))))


imp.rpm.justcalls <- imp.rpm.all[,c(1,19:22)]
imp.rpm.onlyimps <- subset(imp.rpm.justcalls, imp.rpm.justcalls$exp11 != "biparental" | imp.rpm.justcalls$exp17 != "biparental" |imp.rpm.justcalls$exp14 != "biparental" | imp.rpm.justcalls$exp21 != "biparental")

alluvial_DE_gene <- ggplot(data = imp.rpm.onlyimps, aes(axis1 = exp11, axis3 = exp14, axis2 = exp17, axis4 = exp21)) + geom_alluvium(aes(fill = exp14),curve_type = "cubic") + geom_stratum() + geom_text(stat = "stratum",aes(label = after_stat(stratum))) + coord_flip() + ggtitle("imprints over time") + theme_set(theme_void() + theme(legend.position = "bottom"))

alluvial_DE_gene

imp.rpm.onlyimps$pattern <- paste(imp.rpm.onlyimps$exp11, "-", imp.rpm.onlyimps$exp14, "-",imp.rpm.onlyimps$exp17,"-",imp.rpm.onlyimps$exp21)

imprinting_patterns <- imp.rpm.onlyimps %>%
  group_by(pattern) %>%
  summarise(n=n())%>%
  mutate(Freq=n/sum(n))

mat_only_pattern <- subset(imp.rpm.onlyimps, imp.rpm.onlyimps$pattern == "maternal - maternal - maternal - maternal")

B73_mat_only <- subset(mat_only_pattern$feature, grepl("^Zm00001eb", mat_only_pattern$feature))
W22_mat_only <- subset(mat_only_pattern$feature, grepl("^Zm00004", mat_only_pattern$feature))

```

